// @storefront-ui/react/dist/components/SfLoaderCircular/SfLoaderCircular.mjs
import { jsx as s } from "react/jsx-runtime";
import i from "classnames";
import { SfLoaderSize as r } from "@storefront-ui/shared";
const t = {
  [r.xs]: "h-4 w-4 ring-2",
  [r.sm]: "h-5 w-5 ring-2",
  [r.base]: "h-6 w-6 ring-2",
  [r.lg]: "h-8 w-8 ring-2",
  [r.xl]: "h-10 w-10 ring-2",
  [r["2xl"]]: "h-14 w-14 ring-[3px]",
  [r["3xl"]]: "h-24 w-24 ring-4",
  [r["4xl"]]: "h-48 w-48 ring-8"
}, n = {
  [r.xs]: "stroke-[10px]",
  [r.sm]: "stroke-[8px]",
  [r.base]: "stroke-[6px]",
  [r.lg]: "stroke-[4px]",
  [r.xl]: "stroke-[3px]",
  [r["2xl"]]: "stroke-[3px]",
  [r["3xl"]]: "stroke-2",
  [r["4xl"]]: "stroke-2"
};
function m({
  size: e = r.base,
  ariaLabel: l = "loading",
  className: a,
  ...o
}) {
  return /* @__PURE__ */ s(
    "svg",
    {
      className: i(
        "inline-block ring-inset ring-neutral-300 text-primary-700 rounded-full animate-spin-slow",
        t[e],
        a
      ),
      viewBox: "25 25 50 50",
      "aria-live": "polite",
      "aria-label": l,
      "data-testid": "loader-circular",
      ...o,
      children: /* @__PURE__ */ s(
        "circle",
        {
          className: i(
            "fill-none stroke-2 stroke-current animate-stroke-loader-circular",
            n[e]
          ),
          cx: "50",
          cy: "50",
          r: "24"
        }
      )
    }
  );
}
export {
  m as default
};
