// @storefront-ui/typography/dist/index.mjs
var g = {}, s = {};
(function(i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), Object.defineProperty(i, "default", {
    enumerable: !0,
    get: () => l
  });
  function n(e, t) {
    return {
      handler: e,
      config: t
    };
  }
  n.withOptions = function(e, t = () => ({})) {
    const o = function(f) {
      return {
        __options: f,
        handler: e(f),
        config: t(f)
      };
    };
    return o.__isOptionsFunction = !0, o.__pluginFunction = e, o.__configFunction = t, o;
  };
  const l = n;
})(s);
(function(i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), Object.defineProperty(i, "default", {
    enumerable: !0,
    get: () => e
  });
  const n = /* @__PURE__ */ l(s);
  function l(t) {
    return t && t.__esModule ? t : {
      default: t
    };
  }
  const e = n.default;
})(g);
let a = g;
var h = (a.__esModule ? a : { default: a }).default;
const r = "sfTypography", u = h.withOptions(
  ({ utilityPrefix: i = "typography" } = {}) => function({ matchUtilities: n, theme: l }) {
    n(
      {
        [i]: (e) => {
          if (Array.isArray(e.fontSize)) {
            const t = e.fontSize[1];
            t && (e = {
              ...typeof t == "object" ? t : { lineHeight: t },
              ...e
            }), e.fontSize = e.fontSize[0];
          }
          return { ...e };
        }
      },
      {
        values: l(r),
        supportsNegativeValues: !1
      }
    );
  },
  () => ({
    theme: {
      [r]: ({ theme: i }) => [
        // [name, fontSize, lineHeight, fontFamily]
        ["headline-1", "fontSize.6xl", "lineHeight.extra-tight", "fontFamily.headings"],
        ["headline-2", "fontSize.4xl", "lineHeight.10", "fontFamily.headings"],
        ["headline-3", "fontSize.2xl", "lineHeight.8", "fontFamily.headings"],
        ["headline-4", "fontSize.lg", "lineHeight.7", "fontFamily.headings"],
        ["headline-5", "fontSize.base", "lineHeight.6", "fontFamily.headings"],
        ["headline-6", "fontSize.sm", "lineHeight.5", "fontFamily.headings"],
        ["text-xl", "fontSize.xl", "lineHeight.7"],
        ["text-lg", "fontSize.lg", "lineHeight.7"],
        ["text-base", "fontSize.base", "lineHeight.6"],
        ["text-sm", "fontSize.sm", "lineHeight.5"],
        ["text-xs", "fontSize.xs", "lineHeight.4"],
        ["button-lg", "fontSize.lg", "lineHeight.7"],
        ["button-base", "fontSize.base", "lineHeight.6"],
        ["button-sm", "fontSize.sm", "lineHeight.5"],
        ["label-lg", "fontSize.lg", "lineHeight.7"],
        ["label-base", "fontSize.base", "lineHeight.6"],
        ["label-sm", "fontSize.sm", "lineHeight.5"],
        ["error-lg", "fontSize.lg", "lineHeight.7"],
        ["error-base", "fontSize.base", "lineHeight.6"],
        ["error-sm", "fontSize.sm", "lineHeight.5"],
        ["hint-lg", "fontSize.lg", "lineHeight.7"],
        ["hint-base", "fontSize.base", "lineHeight.6"],
        ["hint-sm", "fontSize.sm", "lineHeight.5"]
      ].reduce((n, [l, e, t, o]) => (n[l] = {
        fontSize: i(e),
        lineHeight: i(t),
        fontFamily: o ? i(o) : void 0
      }, n), {})
    }
  })
);
export {
  u as default
};
