// @storefront-ui/tailwind-config/dist/index.mjs
import Wn from "@mertasan/tailwindcss-variables";
import Un from "@storefront-ui/tw-plugin-peer-next";
var Ue = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function $n(a) {
  return a && a.__esModule && Object.prototype.hasOwnProperty.call(a, "default") ? a.default : a;
}
var cr = {}, fr = {};
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), Object.defineProperty(a, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  function t(p, v) {
    return {
      handler: p,
      config: v
    };
  }
  t.withOptions = function(p, v = () => ({})) {
    const y = function(f) {
      return {
        __options: f,
        handler: p(f),
        config: v(f)
      };
    };
    return y.__isOptionsFunction = !0, y.__pluginFunction = p, y.__configFunction = v, y;
  };
  const b = t;
})(fr);
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), Object.defineProperty(a, "default", {
    enumerable: !0,
    get: function() {
      return p;
    }
  });
  const t = /* @__PURE__ */ b(fr);
  function b(v) {
    return v && v.__esModule ? v : {
      default: v
    };
  }
  const p = t.default;
})(cr);
let ht = cr;
var Hn = (ht.__esModule ? ht : { default: ht }).default, et = { exports: {} };
et.exports;
(function(a, t) {
  var b = 200, p = "__lodash_hash_undefined__", v = 800, y = 16, f = 9007199254740991, g = "[object Arguments]", n = "[object Array]", o = "[object AsyncFunction]", l = "[object Boolean]", c = "[object Date]", w = "[object Error]", B = "[object Function]", F = "[object GeneratorFunction]", i = "[object Map]", d = "[object Number]", S = "[object Null]", O = "[object Object]", E = "[object Proxy]", k = "[object RegExp]", T = "[object Set]", x = "[object String]", N = "[object Undefined]", L = "[object WeakMap]", A = "[object ArrayBuffer]", D = "[object DataView]", Y = "[object Float32Array]", Z = "[object Float64Array]", pe = "[object Int8Array]", ue = "[object Int16Array]", ce = "[object Int32Array]", j = "[object Uint8Array]", he = "[object Uint8ClampedArray]", Se = "[object Uint16Array]", Fe = "[object Uint32Array]", Ee = /[\\^$.*+?()[\]{}|]/g, q = /^\[object .+?Constructor\]$/, R = /^(?:0|[1-9]\d*)$/, I = {};
  I[Y] = I[Z] = I[pe] = I[ue] = I[ce] = I[j] = I[he] = I[Se] = I[Fe] = !0, I[g] = I[n] = I[A] = I[l] = I[D] = I[c] = I[w] = I[B] = I[i] = I[d] = I[O] = I[k] = I[T] = I[x] = I[L] = !1;
  var m = typeof Ue == "object" && Ue && Ue.Object === Object && Ue, P = typeof self == "object" && self && self.Object === Object && self, z = m || P || Function("return this")(), _ = t && !t.nodeType && t, $ = _ && !0 && a && !a.nodeType && a, K = $ && $.exports === _, Q = K && m.process, W = function() {
    try {
      var r = $ && $.require && $.require("util").types;
      return r || Q && Q.binding && Q.binding("util");
    } catch {
    }
  }(), ee = W && W.isTypedArray;
  function le(r, u, h) {
    switch (h.length) {
      case 0:
        return r.call(u);
      case 1:
        return r.call(u, h[0]);
      case 2:
        return r.call(u, h[0], h[1]);
      case 3:
        return r.call(u, h[0], h[1], h[2]);
    }
    return r.apply(u, h);
  }
  function H(r, u) {
    for (var h = -1, C = Array(r); ++h < r; )
      C[h] = u(h);
    return C;
  }
  function ie(r) {
    return function(u) {
      return r(u);
    };
  }
  function xe(r, u) {
    return r == null ? void 0 : r[u];
  }
  function Oe(r, u) {
    return function(h) {
      return r(u(h));
    };
  }
  var ge = Array.prototype, fe = Function.prototype, de = Object.prototype, ve = z["__core-js_shared__"], be = fe.toString, _e = de.hasOwnProperty, qe = function() {
    var r = /[^.]+$/.exec(ve && ve.keys && ve.keys.IE_PROTO || "");
    return r ? "Symbol(src)_1." + r : "";
  }(), De = de.toString, rt = be.call(Object), nt = RegExp(
    "^" + be.call(_e).replace(Ee, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
  ), Ne = K ? z.Buffer : void 0, je = z.Symbol, Qe = z.Uint8Array;
  Ne && Ne.allocUnsafe;
  var Ye = Oe(Object.getPrototypeOf, Object), Vt = Object.create, Mr = de.propertyIsEnumerable, Rr = ge.splice, Me = je ? je.toStringTag : void 0, Ve = function() {
    try {
      var r = at(Object, "defineProperty");
      return r({}, "", {}), r;
    } catch {
    }
  }(), Cr = Ne ? Ne.isBuffer : void 0, Kt = Math.max, Nr = Date.now, Jt = at(z, "Map"), $e = at(Object, "create"), zr = /* @__PURE__ */ function() {
    function r() {
    }
    return function(u) {
      if (!Ce(u))
        return {};
      if (Vt)
        return Vt(u);
      r.prototype = u;
      var h = new r();
      return r.prototype = void 0, h;
    };
  }();
  function Re(r) {
    var u = -1, h = r == null ? 0 : r.length;
    for (this.clear(); ++u < h; ) {
      var C = r[u];
      this.set(C[0], C[1]);
    }
  }
  function Wr() {
    this.__data__ = $e ? $e(null) : {}, this.size = 0;
  }
  function Ur(r) {
    var u = this.has(r) && delete this.__data__[r];
    return this.size -= u ? 1 : 0, u;
  }
  function $r(r) {
    var u = this.__data__;
    if ($e) {
      var h = u[r];
      return h === p ? void 0 : h;
    }
    return _e.call(u, r) ? u[r] : void 0;
  }
  function Hr(r) {
    var u = this.__data__;
    return $e ? u[r] !== void 0 : _e.call(u, r);
  }
  function Gr(r, u) {
    var h = this.__data__;
    return this.size += this.has(r) ? 0 : 1, h[r] = $e && u === void 0 ? p : u, this;
  }
  Re.prototype.clear = Wr, Re.prototype.delete = Ur, Re.prototype.get = $r, Re.prototype.has = Hr, Re.prototype.set = Gr;
  function Be(r) {
    var u = -1, h = r == null ? 0 : r.length;
    for (this.clear(); ++u < h; ) {
      var C = r[u];
      this.set(C[0], C[1]);
    }
  }
  function jr() {
    this.__data__ = [], this.size = 0;
  }
  function Qr(r) {
    var u = this.__data__, h = Ke(u, r);
    if (h < 0)
      return !1;
    var C = u.length - 1;
    return h == C ? u.pop() : Rr.call(u, h, 1), --this.size, !0;
  }
  function Yr(r) {
    var u = this.__data__, h = Ke(u, r);
    return h < 0 ? void 0 : u[h][1];
  }
  function Vr(r) {
    return Ke(this.__data__, r) > -1;
  }
  function Kr(r, u) {
    var h = this.__data__, C = Ke(h, r);
    return C < 0 ? (++this.size, h.push([r, u])) : h[C][1] = u, this;
  }
  Be.prototype.clear = jr, Be.prototype.delete = Qr, Be.prototype.get = Yr, Be.prototype.has = Vr, Be.prototype.set = Kr;
  function ze(r) {
    var u = -1, h = r == null ? 0 : r.length;
    for (this.clear(); ++u < h; ) {
      var C = r[u];
      this.set(C[0], C[1]);
    }
  }
  function Jr() {
    this.size = 0, this.__data__ = {
      hash: new Re(),
      map: new (Jt || Be)(),
      string: new Re()
    };
  }
  function Xr(r) {
    var u = Xe(this, r).delete(r);
    return this.size -= u ? 1 : 0, u;
  }
  function Zr(r) {
    return Xe(this, r).get(r);
  }
  function en(r) {
    return Xe(this, r).has(r);
  }
  function tn(r, u) {
    var h = Xe(this, r), C = h.size;
    return h.set(r, u), this.size += h.size == C ? 0 : 1, this;
  }
  ze.prototype.clear = Jr, ze.prototype.delete = Xr, ze.prototype.get = Zr, ze.prototype.has = en, ze.prototype.set = tn;
  function We(r) {
    var u = this.__data__ = new Be(r);
    this.size = u.size;
  }
  function rn() {
    this.__data__ = new Be(), this.size = 0;
  }
  function nn(r) {
    var u = this.__data__, h = u.delete(r);
    return this.size = u.size, h;
  }
  function on(r) {
    return this.__data__.get(r);
  }
  function an(r) {
    return this.__data__.has(r);
  }
  function sn(r, u) {
    var h = this.__data__;
    if (h instanceof Be) {
      var C = h.__data__;
      if (!Jt || C.length < b - 1)
        return C.push([r, u]), this.size = ++h.size, this;
      h = this.__data__ = new ze(C);
    }
    return h.set(r, u), this.size = h.size, this;
  }
  We.prototype.clear = rn, We.prototype.delete = nn, We.prototype.get = on, We.prototype.has = an, We.prototype.set = sn;
  function un(r, u) {
    var h = lt(r), C = !h && ut(r), V = !h && !C && rr(r), te = !h && !C && !V && ir(r), oe = h || C || V || te, J = oe ? H(r.length, String) : [], ae = J.length;
    for (var Pe in r)
      oe && // Safari 9 has enumerable `arguments.length` in strict mode.
      (Pe == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
      V && (Pe == "offset" || Pe == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
      te && (Pe == "buffer" || Pe == "byteLength" || Pe == "byteOffset") || // Skip index properties.
      er(Pe, ae)) || J.push(Pe);
    return J;
  }
  function it(r, u, h) {
    (h !== void 0 && !Ze(r[u], h) || h === void 0 && !(u in r)) && ot(r, u, h);
  }
  function ln(r, u, h) {
    var C = r[u];
    (!(_e.call(r, u) && Ze(C, h)) || h === void 0 && !(u in r)) && ot(r, u, h);
  }
  function Ke(r, u) {
    for (var h = r.length; h--; )
      if (Ze(r[h][0], u))
        return h;
    return -1;
  }
  function ot(r, u, h) {
    u == "__proto__" && Ve ? Ve(r, u, {
      configurable: !0,
      enumerable: !0,
      value: h,
      writable: !0
    }) : r[u] = h;
  }
  var cn = Sn();
  function Je(r) {
    return r == null ? r === void 0 ? N : S : Me && Me in Object(r) ? En(r) : An(r);
  }
  function Xt(r) {
    return He(r) && Je(r) == g;
  }
  function fn(r) {
    if (!Ce(r) || xn(r))
      return !1;
    var u = ft(r) ? nt : q;
    return u.test(Fn(r));
  }
  function pn(r) {
    return He(r) && nr(r.length) && !!I[Je(r)];
  }
  function dn(r) {
    if (!Ce(r))
      return Pn(r);
    var u = tr(r), h = [];
    for (var C in r)
      C == "constructor" && (u || !_e.call(r, C)) || h.push(C);
    return h;
  }
  function Zt(r, u, h, C, V) {
    r !== u && cn(u, function(te, oe) {
      if (V || (V = new We()), Ce(te))
        hn(r, u, oe, h, Zt, C, V);
      else {
        var J = C ? C(st(r, oe), te, oe + "", r, u, V) : void 0;
        J === void 0 && (J = te), it(r, oe, J);
      }
    }, or);
  }
  function hn(r, u, h, C, V, te, oe) {
    var J = st(r, h), ae = st(u, h), Pe = oe.get(ae);
    if (Pe) {
      it(r, h, Pe);
      return;
    }
    var Te = te ? te(J, ae, h + "", r, u, oe) : void 0, Ge = Te === void 0;
    if (Ge) {
      var pt = lt(ae), dt = !pt && rr(ae), sr = !pt && !dt && ir(ae);
      Te = ae, pt || dt || sr ? lt(J) ? Te = J : qn(J) ? Te = _n(J) : dt ? (Ge = !1, Te = mn(ae)) : sr ? (Ge = !1, Te = bn(ae)) : Te = [] : Mn(ae) || ut(ae) ? (Te = J, ut(J) ? Te = Rn(J) : (!Ce(J) || ft(J)) && (Te = On(ae))) : Ge = !1;
    }
    Ge && (oe.set(ae, Te), V(Te, ae, C, te, oe), oe.delete(ae)), it(r, h, Te);
  }
  function gn(r, u) {
    return Dn(Bn(r, u, ar), r + "");
  }
  var vn = Ve ? function(r, u) {
    return Ve(r, "toString", {
      configurable: !0,
      enumerable: !1,
      value: Nn(u),
      writable: !0
    });
  } : ar;
  function mn(r, u) {
    return r.slice();
  }
  function wn(r) {
    var u = new r.constructor(r.byteLength);
    return new Qe(u).set(new Qe(r)), u;
  }
  function bn(r, u) {
    var h = wn(r.buffer);
    return new r.constructor(h, r.byteOffset, r.length);
  }
  function _n(r, u) {
    var h = -1, C = r.length;
    for (u || (u = Array(C)); ++h < C; )
      u[h] = r[h];
    return u;
  }
  function Tn(r, u, h, C) {
    var V = !h;
    h || (h = {});
    for (var te = -1, oe = u.length; ++te < oe; ) {
      var J = u[te], ae = void 0;
      ae === void 0 && (ae = r[J]), V ? ot(h, J, ae) : ln(h, J, ae);
    }
    return h;
  }
  function yn(r) {
    return gn(function(u, h) {
      var C = -1, V = h.length, te = V > 1 ? h[V - 1] : void 0, oe = V > 2 ? h[2] : void 0;
      for (te = r.length > 3 && typeof te == "function" ? (V--, te) : void 0, oe && In(h[0], h[1], oe) && (te = V < 3 ? void 0 : te, V = 1), u = Object(u); ++C < V; ) {
        var J = h[C];
        J && r(u, J, C, te);
      }
      return u;
    });
  }
  function Sn(r) {
    return function(u, h, C) {
      for (var V = -1, te = Object(u), oe = C(u), J = oe.length; J--; ) {
        var ae = oe[++V];
        if (h(te[ae], ae, te) === !1)
          break;
      }
      return u;
    };
  }
  function Xe(r, u) {
    var h = r.__data__;
    return kn(u) ? h[typeof u == "string" ? "string" : "hash"] : h.map;
  }
  function at(r, u) {
    var h = xe(r, u);
    return fn(h) ? h : void 0;
  }
  function En(r) {
    var u = _e.call(r, Me), h = r[Me];
    try {
      r[Me] = void 0;
      var C = !0;
    } catch {
    }
    var V = De.call(r);
    return C && (u ? r[Me] = h : delete r[Me]), V;
  }
  function On(r) {
    return typeof r.constructor == "function" && !tr(r) ? zr(Ye(r)) : {};
  }
  function er(r, u) {
    var h = typeof r;
    return u = u ?? f, !!u && (h == "number" || h != "symbol" && R.test(r)) && r > -1 && r % 1 == 0 && r < u;
  }
  function In(r, u, h) {
    if (!Ce(h))
      return !1;
    var C = typeof u;
    return (C == "number" ? ct(h) && er(u, h.length) : C == "string" && u in h) ? Ze(h[u], r) : !1;
  }
  function kn(r) {
    var u = typeof r;
    return u == "string" || u == "number" || u == "symbol" || u == "boolean" ? r !== "__proto__" : r === null;
  }
  function xn(r) {
    return !!qe && qe in r;
  }
  function tr(r) {
    var u = r && r.constructor, h = typeof u == "function" && u.prototype || de;
    return r === h;
  }
  function Pn(r) {
    var u = [];
    if (r != null)
      for (var h in Object(r))
        u.push(h);
    return u;
  }
  function An(r) {
    return De.call(r);
  }
  function Bn(r, u, h) {
    return u = Kt(u === void 0 ? r.length - 1 : u, 0), function() {
      for (var C = arguments, V = -1, te = Kt(C.length - u, 0), oe = Array(te); ++V < te; )
        oe[V] = C[u + V];
      V = -1;
      for (var J = Array(u + 1); ++V < u; )
        J[V] = C[V];
      return J[u] = h(oe), le(r, this, J);
    };
  }
  function st(r, u) {
    if (!(u === "constructor" && typeof r[u] == "function") && u != "__proto__")
      return r[u];
  }
  var Dn = Ln(vn);
  function Ln(r) {
    var u = 0, h = 0;
    return function() {
      var C = Nr(), V = y - (C - h);
      if (h = C, V > 0) {
        if (++u >= v)
          return arguments[0];
      } else
        u = 0;
      return r.apply(void 0, arguments);
    };
  }
  function Fn(r) {
    if (r != null) {
      try {
        return be.call(r);
      } catch {
      }
      try {
        return r + "";
      } catch {
      }
    }
    return "";
  }
  function Ze(r, u) {
    return r === u || r !== r && u !== u;
  }
  var ut = Xt(/* @__PURE__ */ function() {
    return arguments;
  }()) ? Xt : function(r) {
    return He(r) && _e.call(r, "callee") && !Mr.call(r, "callee");
  }, lt = Array.isArray;
  function ct(r) {
    return r != null && nr(r.length) && !ft(r);
  }
  function qn(r) {
    return He(r) && ct(r);
  }
  var rr = Cr || zn;
  function ft(r) {
    if (!Ce(r))
      return !1;
    var u = Je(r);
    return u == B || u == F || u == o || u == E;
  }
  function nr(r) {
    return typeof r == "number" && r > -1 && r % 1 == 0 && r <= f;
  }
  function Ce(r) {
    var u = typeof r;
    return r != null && (u == "object" || u == "function");
  }
  function He(r) {
    return r != null && typeof r == "object";
  }
  function Mn(r) {
    if (!He(r) || Je(r) != O)
      return !1;
    var u = Ye(r);
    if (u === null)
      return !0;
    var h = _e.call(u, "constructor") && u.constructor;
    return typeof h == "function" && h instanceof h && be.call(h) == rt;
  }
  var ir = ee ? ie(ee) : pn;
  function Rn(r) {
    return Tn(r, or(r));
  }
  function or(r) {
    return ct(r) ? un(r) : dn(r);
  }
  var Cn = yn(function(r, u, h) {
    Zt(r, u, h);
  });
  function Nn(r) {
    return function() {
      return r;
    };
  }
  function ar(r) {
    return r;
  }
  function zn() {
    return !1;
  }
  a.exports = Cn;
})(et, et.exports);
var Gn = et.exports;
function jn() {
  if (!arguments.length)
    return [];
  var a = arguments[0];
  return Qn(a) ? a : [a];
}
var Qn = Array.isArray, Yn = jn, pr = {}, dr = {}, Wt = { exports: {} }, ne = String, hr = function() {
  return { isColorSupported: !1, reset: ne, bold: ne, dim: ne, italic: ne, underline: ne, inverse: ne, hidden: ne, strikethrough: ne, black: ne, red: ne, green: ne, yellow: ne, blue: ne, magenta: ne, cyan: ne, white: ne, gray: ne, bgBlack: ne, bgRed: ne, bgGreen: ne, bgYellow: ne, bgBlue: ne, bgMagenta: ne, bgCyan: ne, bgWhite: ne };
};
Wt.exports = hr();
Wt.exports.createColors = hr;
var Vn = Wt.exports;
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  });
  function t(g, n) {
    for (var o in n)
      Object.defineProperty(g, o, {
        enumerable: !0,
        get: n[o]
      });
  }
  t(a, {
    dim: function() {
      return y;
    },
    default: function() {
      return f;
    }
  });
  const b = /* @__PURE__ */ p(Vn);
  function p(g) {
    return g && g.__esModule ? g : {
      default: g
    };
  }
  function v(g, n, o) {
    typeof process < "u" && process.env.JEST_WORKER_ID || (console.warn(""), n.forEach((l) => console.warn(g, "-", l)));
  }
  function y(g) {
    return b.default.dim(g);
  }
  const f = {
    info(g, n) {
      v(b.default.bold(b.default.cyan("info")), ...Array.isArray(g) ? [
        g
      ] : [
        n,
        g
      ]);
    },
    warn(g, n) {
      v(b.default.bold(b.default.yellow("warn")), ...Array.isArray(g) ? [
        g
      ] : [
        n,
        g
      ]);
    },
    risk(g, n) {
      v(b.default.bold(b.default.magenta("risk")), ...Array.isArray(g) ? [
        g
      ] : [
        n,
        g
      ]);
    }
  };
})(dr);
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), Object.defineProperty(a, "default", {
    enumerable: !0,
    get: function() {
      return v;
    }
  });
  const t = /* @__PURE__ */ b(dr);
  function b(y) {
    return y && y.__esModule ? y : {
      default: y
    };
  }
  function p({ version: y, from: f, to: g }) {
    t.default.warn(`${f}-color-renamed`, [
      `As of Tailwind CSS ${y}, \`${f}\` has been renamed to \`${g}\`.`,
      "Update your configuration file to silence this warning."
    ]);
  }
  const v = {
    inherit: "inherit",
    current: "currentColor",
    transparent: "transparent",
    black: "#000",
    white: "#fff",
    slate: {
      50: "#f8fafc",
      100: "#f1f5f9",
      200: "#e2e8f0",
      300: "#cbd5e1",
      400: "#94a3b8",
      500: "#64748b",
      600: "#475569",
      700: "#334155",
      800: "#1e293b",
      900: "#0f172a",
      950: "#020617"
    },
    gray: {
      50: "#f9fafb",
      100: "#f3f4f6",
      200: "#e5e7eb",
      300: "#d1d5db",
      400: "#9ca3af",
      500: "#6b7280",
      600: "#4b5563",
      700: "#374151",
      800: "#1f2937",
      900: "#111827",
      950: "#030712"
    },
    zinc: {
      50: "#fafafa",
      100: "#f4f4f5",
      200: "#e4e4e7",
      300: "#d4d4d8",
      400: "#a1a1aa",
      500: "#71717a",
      600: "#52525b",
      700: "#3f3f46",
      800: "#27272a",
      900: "#18181b",
      950: "#09090b"
    },
    neutral: {
      50: "#fafafa",
      100: "#f5f5f5",
      200: "#e5e5e5",
      300: "#d4d4d4",
      400: "#a3a3a3",
      500: "#737373",
      600: "#525252",
      700: "#404040",
      800: "#262626",
      900: "#171717",
      950: "#0a0a0a"
    },
    stone: {
      50: "#fafaf9",
      100: "#f5f5f4",
      200: "#e7e5e4",
      300: "#d6d3d1",
      400: "#a8a29e",
      500: "#78716c",
      600: "#57534e",
      700: "#44403c",
      800: "#292524",
      900: "#1c1917",
      950: "#0c0a09"
    },
    red: {
      50: "#fef2f2",
      100: "#fee2e2",
      200: "#fecaca",
      300: "#fca5a5",
      400: "#f87171",
      500: "#ef4444",
      600: "#dc2626",
      700: "#b91c1c",
      800: "#991b1b",
      900: "#7f1d1d",
      950: "#450a0a"
    },
    orange: {
      50: "#fff7ed",
      100: "#ffedd5",
      200: "#fed7aa",
      300: "#fdba74",
      400: "#fb923c",
      500: "#f97316",
      600: "#ea580c",
      700: "#c2410c",
      800: "#9a3412",
      900: "#7c2d12",
      950: "#431407"
    },
    amber: {
      50: "#fffbeb",
      100: "#fef3c7",
      200: "#fde68a",
      300: "#fcd34d",
      400: "#fbbf24",
      500: "#f59e0b",
      600: "#d97706",
      700: "#b45309",
      800: "#92400e",
      900: "#78350f",
      950: "#451a03"
    },
    yellow: {
      50: "#fefce8",
      100: "#fef9c3",
      200: "#fef08a",
      300: "#fde047",
      400: "#facc15",
      500: "#eab308",
      600: "#ca8a04",
      700: "#a16207",
      800: "#854d0e",
      900: "#713f12",
      950: "#422006"
    },
    lime: {
      50: "#f7fee7",
      100: "#ecfccb",
      200: "#d9f99d",
      300: "#bef264",
      400: "#a3e635",
      500: "#84cc16",
      600: "#65a30d",
      700: "#4d7c0f",
      800: "#3f6212",
      900: "#365314",
      950: "#1a2e05"
    },
    green: {
      50: "#f0fdf4",
      100: "#dcfce7",
      200: "#bbf7d0",
      300: "#86efac",
      400: "#4ade80",
      500: "#22c55e",
      600: "#16a34a",
      700: "#15803d",
      800: "#166534",
      900: "#14532d",
      950: "#052e16"
    },
    emerald: {
      50: "#ecfdf5",
      100: "#d1fae5",
      200: "#a7f3d0",
      300: "#6ee7b7",
      400: "#34d399",
      500: "#10b981",
      600: "#059669",
      700: "#047857",
      800: "#065f46",
      900: "#064e3b",
      950: "#022c22"
    },
    teal: {
      50: "#f0fdfa",
      100: "#ccfbf1",
      200: "#99f6e4",
      300: "#5eead4",
      400: "#2dd4bf",
      500: "#14b8a6",
      600: "#0d9488",
      700: "#0f766e",
      800: "#115e59",
      900: "#134e4a",
      950: "#042f2e"
    },
    cyan: {
      50: "#ecfeff",
      100: "#cffafe",
      200: "#a5f3fc",
      300: "#67e8f9",
      400: "#22d3ee",
      500: "#06b6d4",
      600: "#0891b2",
      700: "#0e7490",
      800: "#155e75",
      900: "#164e63",
      950: "#083344"
    },
    sky: {
      50: "#f0f9ff",
      100: "#e0f2fe",
      200: "#bae6fd",
      300: "#7dd3fc",
      400: "#38bdf8",
      500: "#0ea5e9",
      600: "#0284c7",
      700: "#0369a1",
      800: "#075985",
      900: "#0c4a6e",
      950: "#082f49"
    },
    blue: {
      50: "#eff6ff",
      100: "#dbeafe",
      200: "#bfdbfe",
      300: "#93c5fd",
      400: "#60a5fa",
      500: "#3b82f6",
      600: "#2563eb",
      700: "#1d4ed8",
      800: "#1e40af",
      900: "#1e3a8a",
      950: "#172554"
    },
    indigo: {
      50: "#eef2ff",
      100: "#e0e7ff",
      200: "#c7d2fe",
      300: "#a5b4fc",
      400: "#818cf8",
      500: "#6366f1",
      600: "#4f46e5",
      700: "#4338ca",
      800: "#3730a3",
      900: "#312e81",
      950: "#1e1b4b"
    },
    violet: {
      50: "#f5f3ff",
      100: "#ede9fe",
      200: "#ddd6fe",
      300: "#c4b5fd",
      400: "#a78bfa",
      500: "#8b5cf6",
      600: "#7c3aed",
      700: "#6d28d9",
      800: "#5b21b6",
      900: "#4c1d95",
      950: "#2e1065"
    },
    purple: {
      50: "#faf5ff",
      100: "#f3e8ff",
      200: "#e9d5ff",
      300: "#d8b4fe",
      400: "#c084fc",
      500: "#a855f7",
      600: "#9333ea",
      700: "#7e22ce",
      800: "#6b21a8",
      900: "#581c87",
      950: "#3b0764"
    },
    fuchsia: {
      50: "#fdf4ff",
      100: "#fae8ff",
      200: "#f5d0fe",
      300: "#f0abfc",
      400: "#e879f9",
      500: "#d946ef",
      600: "#c026d3",
      700: "#a21caf",
      800: "#86198f",
      900: "#701a75",
      950: "#4a044e"
    },
    pink: {
      50: "#fdf2f8",
      100: "#fce7f3",
      200: "#fbcfe8",
      300: "#f9a8d4",
      400: "#f472b6",
      500: "#ec4899",
      600: "#db2777",
      700: "#be185d",
      800: "#9d174d",
      900: "#831843",
      950: "#500724"
    },
    rose: {
      50: "#fff1f2",
      100: "#ffe4e6",
      200: "#fecdd3",
      300: "#fda4af",
      400: "#fb7185",
      500: "#f43f5e",
      600: "#e11d48",
      700: "#be123c",
      800: "#9f1239",
      900: "#881337",
      950: "#4c0519"
    },
    get lightBlue() {
      return p({
        version: "v2.2",
        from: "lightBlue",
        to: "sky"
      }), this.sky;
    },
    get warmGray() {
      return p({
        version: "v3.0",
        from: "warmGray",
        to: "stone"
      }), this.stone;
    },
    get trueGray() {
      return p({
        version: "v3.0",
        from: "trueGray",
        to: "neutral"
      }), this.neutral;
    },
    get coolGray() {
      return p({
        version: "v3.0",
        from: "coolGray",
        to: "gray"
      }), this.gray;
    },
    get blueGray() {
      return p({
        version: "v3.0",
        from: "blueGray",
        to: "slate"
      }), this.slate;
    }
  };
})(pr);
let gt = pr;
var Kn = (gt.__esModule ? gt : { default: gt }).default;
const s = Kn, U = (a) => a.toFixed(7).replace(/(\.[0-9]+?)0+$/, "$1").replace(/\.0$/, ""), me = (a) => `${U(a / 16)}rem`, e = (a, t) => `${U(a / t)}em`, Ae = (a) => {
  a = a.replace("#", ""), a = a.length === 3 ? a.replace(/./g, "$&$&") : a;
  const t = parseInt(a.substring(0, 2), 16), b = parseInt(a.substring(2, 4), 16), p = parseInt(a.substring(4, 6), 16);
  return `${t} ${b} ${p}`;
};
let vt = {
  sm: {
    css: [
      {
        fontSize: me(14),
        lineHeight: U(24 / 14),
        p: {
          marginTop: e(16, 14),
          marginBottom: e(16, 14)
        },
        '[class~="lead"]': {
          fontSize: e(18, 14),
          lineHeight: U(28 / 18),
          marginTop: e(16, 18),
          marginBottom: e(16, 18)
        },
        blockquote: {
          marginTop: e(24, 18),
          marginBottom: e(24, 18),
          paddingInlineStart: e(20, 18)
        },
        h1: {
          fontSize: e(30, 14),
          marginTop: "0",
          marginBottom: e(24, 30),
          lineHeight: U(36 / 30)
        },
        h2: {
          fontSize: e(20, 14),
          marginTop: e(32, 20),
          marginBottom: e(16, 20),
          lineHeight: U(28 / 20)
        },
        h3: {
          fontSize: e(18, 14),
          marginTop: e(28, 18),
          marginBottom: e(8, 18),
          lineHeight: U(28 / 18)
        },
        h4: {
          marginTop: e(20, 14),
          marginBottom: e(8, 14),
          lineHeight: U(20 / 14)
        },
        img: {
          marginTop: e(24, 14),
          marginBottom: e(24, 14)
        },
        picture: {
          marginTop: e(24, 14),
          marginBottom: e(24, 14)
        },
        "picture > img": {
          marginTop: "0",
          marginBottom: "0"
        },
        video: {
          marginTop: e(24, 14),
          marginBottom: e(24, 14)
        },
        kbd: {
          fontSize: e(12, 14),
          borderRadius: me(5),
          paddingTop: e(2, 14),
          paddingInlineEnd: e(5, 14),
          paddingBottom: e(2, 14),
          paddingInlineStart: e(5, 14)
        },
        code: {
          fontSize: e(12, 14)
        },
        "h2 code": {
          fontSize: e(18, 20)
        },
        "h3 code": {
          fontSize: e(16, 18)
        },
        pre: {
          fontSize: e(12, 14),
          lineHeight: U(20 / 12),
          marginTop: e(20, 12),
          marginBottom: e(20, 12),
          borderRadius: me(4),
          paddingTop: e(8, 12),
          paddingInlineEnd: e(12, 12),
          paddingBottom: e(8, 12),
          paddingInlineStart: e(12, 12)
        },
        ol: {
          marginTop: e(16, 14),
          marginBottom: e(16, 14),
          paddingInlineStart: e(22, 14)
        },
        ul: {
          marginTop: e(16, 14),
          marginBottom: e(16, 14),
          paddingInlineStart: e(22, 14)
        },
        li: {
          marginTop: e(4, 14),
          marginBottom: e(4, 14)
        },
        "ol > li": {
          paddingInlineStart: e(6, 14)
        },
        "ul > li": {
          paddingInlineStart: e(6, 14)
        },
        "> ul > li p": {
          marginTop: e(8, 14),
          marginBottom: e(8, 14)
        },
        "> ul > li > p:first-child": {
          marginTop: e(16, 14)
        },
        "> ul > li > p:last-child": {
          marginBottom: e(16, 14)
        },
        "> ol > li > p:first-child": {
          marginTop: e(16, 14)
        },
        "> ol > li > p:last-child": {
          marginBottom: e(16, 14)
        },
        "ul ul, ul ol, ol ul, ol ol": {
          marginTop: e(8, 14),
          marginBottom: e(8, 14)
        },
        dl: {
          marginTop: e(16, 14),
          marginBottom: e(16, 14)
        },
        dt: {
          marginTop: e(16, 14)
        },
        dd: {
          marginTop: e(4, 14),
          paddingInlineStart: e(22, 14)
        },
        hr: {
          marginTop: e(40, 14),
          marginBottom: e(40, 14)
        },
        "hr + *": {
          marginTop: "0"
        },
        "h2 + *": {
          marginTop: "0"
        },
        "h3 + *": {
          marginTop: "0"
        },
        "h4 + *": {
          marginTop: "0"
        },
        table: {
          fontSize: e(12, 14),
          lineHeight: U(18 / 12)
        },
        "thead th": {
          paddingInlineEnd: e(12, 12),
          paddingBottom: e(8, 12),
          paddingInlineStart: e(12, 12)
        },
        "thead th:first-child": {
          paddingInlineStart: "0"
        },
        "thead th:last-child": {
          paddingInlineEnd: "0"
        },
        "tbody td, tfoot td": {
          paddingTop: e(8, 12),
          paddingInlineEnd: e(12, 12),
          paddingBottom: e(8, 12),
          paddingInlineStart: e(12, 12)
        },
        "tbody td:first-child, tfoot td:first-child": {
          paddingInlineStart: "0"
        },
        "tbody td:last-child, tfoot td:last-child": {
          paddingInlineEnd: "0"
        },
        figure: {
          marginTop: e(24, 14),
          marginBottom: e(24, 14)
        },
        "figure > *": {
          marginTop: "0",
          marginBottom: "0"
        },
        figcaption: {
          fontSize: e(12, 14),
          lineHeight: U(16 / 12),
          marginTop: e(8, 12)
        }
      },
      {
        "> :first-child": {
          marginTop: "0"
        },
        "> :last-child": {
          marginBottom: "0"
        }
      }
    ]
  },
  base: {
    css: [
      {
        fontSize: me(16),
        lineHeight: U(28 / 16),
        p: {
          marginTop: e(20, 16),
          marginBottom: e(20, 16)
        },
        '[class~="lead"]': {
          fontSize: e(20, 16),
          lineHeight: U(32 / 20),
          marginTop: e(24, 20),
          marginBottom: e(24, 20)
        },
        blockquote: {
          marginTop: e(32, 20),
          marginBottom: e(32, 20),
          paddingInlineStart: e(20, 20)
        },
        h1: {
          fontSize: e(36, 16),
          marginTop: "0",
          marginBottom: e(32, 36),
          lineHeight: U(40 / 36)
        },
        h2: {
          fontSize: e(24, 16),
          marginTop: e(48, 24),
          marginBottom: e(24, 24),
          lineHeight: U(32 / 24)
        },
        h3: {
          fontSize: e(20, 16),
          marginTop: e(32, 20),
          marginBottom: e(12, 20),
          lineHeight: U(32 / 20)
        },
        h4: {
          marginTop: e(24, 16),
          marginBottom: e(8, 16),
          lineHeight: U(24 / 16)
        },
        img: {
          marginTop: e(32, 16),
          marginBottom: e(32, 16)
        },
        picture: {
          marginTop: e(32, 16),
          marginBottom: e(32, 16)
        },
        "picture > img": {
          marginTop: "0",
          marginBottom: "0"
        },
        video: {
          marginTop: e(32, 16),
          marginBottom: e(32, 16)
        },
        kbd: {
          fontSize: e(14, 16),
          borderRadius: me(5),
          paddingTop: e(3, 16),
          paddingInlineEnd: e(6, 16),
          paddingBottom: e(3, 16),
          paddingInlineStart: e(6, 16)
        },
        code: {
          fontSize: e(14, 16)
        },
        "h2 code": {
          fontSize: e(21, 24)
        },
        "h3 code": {
          fontSize: e(18, 20)
        },
        pre: {
          fontSize: e(14, 16),
          lineHeight: U(24 / 14),
          marginTop: e(24, 14),
          marginBottom: e(24, 14),
          borderRadius: me(6),
          paddingTop: e(12, 14),
          paddingInlineEnd: e(16, 14),
          paddingBottom: e(12, 14),
          paddingInlineStart: e(16, 14)
        },
        ol: {
          marginTop: e(20, 16),
          marginBottom: e(20, 16),
          paddingInlineStart: e(26, 16)
        },
        ul: {
          marginTop: e(20, 16),
          marginBottom: e(20, 16),
          paddingInlineStart: e(26, 16)
        },
        li: {
          marginTop: e(8, 16),
          marginBottom: e(8, 16)
        },
        "ol > li": {
          paddingInlineStart: e(6, 16)
        },
        "ul > li": {
          paddingInlineStart: e(6, 16)
        },
        "> ul > li p": {
          marginTop: e(12, 16),
          marginBottom: e(12, 16)
        },
        "> ul > li > p:first-child": {
          marginTop: e(20, 16)
        },
        "> ul > li > p:last-child": {
          marginBottom: e(20, 16)
        },
        "> ol > li > p:first-child": {
          marginTop: e(20, 16)
        },
        "> ol > li > p:last-child": {
          marginBottom: e(20, 16)
        },
        "ul ul, ul ol, ol ul, ol ol": {
          marginTop: e(12, 16),
          marginBottom: e(12, 16)
        },
        dl: {
          marginTop: e(20, 16),
          marginBottom: e(20, 16)
        },
        dt: {
          marginTop: e(20, 16)
        },
        dd: {
          marginTop: e(8, 16),
          paddingInlineStart: e(26, 16)
        },
        hr: {
          marginTop: e(48, 16),
          marginBottom: e(48, 16)
        },
        "hr + *": {
          marginTop: "0"
        },
        "h2 + *": {
          marginTop: "0"
        },
        "h3 + *": {
          marginTop: "0"
        },
        "h4 + *": {
          marginTop: "0"
        },
        table: {
          fontSize: e(14, 16),
          lineHeight: U(24 / 14)
        },
        "thead th": {
          paddingInlineEnd: e(8, 14),
          paddingBottom: e(8, 14),
          paddingInlineStart: e(8, 14)
        },
        "thead th:first-child": {
          paddingInlineStart: "0"
        },
        "thead th:last-child": {
          paddingInlineEnd: "0"
        },
        "tbody td, tfoot td": {
          paddingTop: e(8, 14),
          paddingInlineEnd: e(8, 14),
          paddingBottom: e(8, 14),
          paddingInlineStart: e(8, 14)
        },
        "tbody td:first-child, tfoot td:first-child": {
          paddingInlineStart: "0"
        },
        "tbody td:last-child, tfoot td:last-child": {
          paddingInlineEnd: "0"
        },
        figure: {
          marginTop: e(32, 16),
          marginBottom: e(32, 16)
        },
        "figure > *": {
          marginTop: "0",
          marginBottom: "0"
        },
        figcaption: {
          fontSize: e(14, 16),
          lineHeight: U(20 / 14),
          marginTop: e(12, 14)
        }
      },
      {
        "> :first-child": {
          marginTop: "0"
        },
        "> :last-child": {
          marginBottom: "0"
        }
      }
    ]
  },
  lg: {
    css: [
      {
        fontSize: me(18),
        lineHeight: U(32 / 18),
        p: {
          marginTop: e(24, 18),
          marginBottom: e(24, 18)
        },
        '[class~="lead"]': {
          fontSize: e(22, 18),
          lineHeight: U(32 / 22),
          marginTop: e(24, 22),
          marginBottom: e(24, 22)
        },
        blockquote: {
          marginTop: e(40, 24),
          marginBottom: e(40, 24),
          paddingInlineStart: e(24, 24)
        },
        h1: {
          fontSize: e(48, 18),
          marginTop: "0",
          marginBottom: e(40, 48),
          lineHeight: U(48 / 48)
        },
        h2: {
          fontSize: e(30, 18),
          marginTop: e(56, 30),
          marginBottom: e(32, 30),
          lineHeight: U(40 / 30)
        },
        h3: {
          fontSize: e(24, 18),
          marginTop: e(40, 24),
          marginBottom: e(16, 24),
          lineHeight: U(36 / 24)
        },
        h4: {
          marginTop: e(32, 18),
          marginBottom: e(8, 18),
          lineHeight: U(28 / 18)
        },
        img: {
          marginTop: e(32, 18),
          marginBottom: e(32, 18)
        },
        picture: {
          marginTop: e(32, 18),
          marginBottom: e(32, 18)
        },
        "picture > img": {
          marginTop: "0",
          marginBottom: "0"
        },
        video: {
          marginTop: e(32, 18),
          marginBottom: e(32, 18)
        },
        kbd: {
          fontSize: e(16, 18),
          borderRadius: me(5),
          paddingTop: e(4, 18),
          paddingInlineEnd: e(8, 18),
          paddingBottom: e(4, 18),
          paddingInlineStart: e(8, 18)
        },
        code: {
          fontSize: e(16, 18)
        },
        "h2 code": {
          fontSize: e(26, 30)
        },
        "h3 code": {
          fontSize: e(21, 24)
        },
        pre: {
          fontSize: e(16, 18),
          lineHeight: U(28 / 16),
          marginTop: e(32, 16),
          marginBottom: e(32, 16),
          borderRadius: me(6),
          paddingTop: e(16, 16),
          paddingInlineEnd: e(24, 16),
          paddingBottom: e(16, 16),
          paddingInlineStart: e(24, 16)
        },
        ol: {
          marginTop: e(24, 18),
          marginBottom: e(24, 18),
          paddingInlineStart: e(28, 18)
        },
        ul: {
          marginTop: e(24, 18),
          marginBottom: e(24, 18),
          paddingInlineStart: e(28, 18)
        },
        li: {
          marginTop: e(12, 18),
          marginBottom: e(12, 18)
        },
        "ol > li": {
          paddingInlineStart: e(8, 18)
        },
        "ul > li": {
          paddingInlineStart: e(8, 18)
        },
        "> ul > li p": {
          marginTop: e(16, 18),
          marginBottom: e(16, 18)
        },
        "> ul > li > p:first-child": {
          marginTop: e(24, 18)
        },
        "> ul > li > p:last-child": {
          marginBottom: e(24, 18)
        },
        "> ol > li > p:first-child": {
          marginTop: e(24, 18)
        },
        "> ol > li > p:last-child": {
          marginBottom: e(24, 18)
        },
        "ul ul, ul ol, ol ul, ol ol": {
          marginTop: e(16, 18),
          marginBottom: e(16, 18)
        },
        dl: {
          marginTop: e(24, 18),
          marginBottom: e(24, 18)
        },
        dt: {
          marginTop: e(24, 18)
        },
        dd: {
          marginTop: e(12, 18),
          paddingInlineStart: e(28, 18)
        },
        hr: {
          marginTop: e(56, 18),
          marginBottom: e(56, 18)
        },
        "hr + *": {
          marginTop: "0"
        },
        "h2 + *": {
          marginTop: "0"
        },
        "h3 + *": {
          marginTop: "0"
        },
        "h4 + *": {
          marginTop: "0"
        },
        table: {
          fontSize: e(16, 18),
          lineHeight: U(24 / 16)
        },
        "thead th": {
          paddingInlineEnd: e(12, 16),
          paddingBottom: e(12, 16),
          paddingInlineStart: e(12, 16)
        },
        "thead th:first-child": {
          paddingInlineStart: "0"
        },
        "thead th:last-child": {
          paddingInlineEnd: "0"
        },
        "tbody td, tfoot td": {
          paddingTop: e(12, 16),
          paddingInlineEnd: e(12, 16),
          paddingBottom: e(12, 16),
          paddingInlineStart: e(12, 16)
        },
        "tbody td:first-child, tfoot td:first-child": {
          paddingInlineStart: "0"
        },
        "tbody td:last-child, tfoot td:last-child": {
          paddingInlineEnd: "0"
        },
        figure: {
          marginTop: e(32, 18),
          marginBottom: e(32, 18)
        },
        "figure > *": {
          marginTop: "0",
          marginBottom: "0"
        },
        figcaption: {
          fontSize: e(16, 18),
          lineHeight: U(24 / 16),
          marginTop: e(16, 16)
        }
      },
      {
        "> :first-child": {
          marginTop: "0"
        },
        "> :last-child": {
          marginBottom: "0"
        }
      }
    ]
  },
  xl: {
    css: [
      {
        fontSize: me(20),
        lineHeight: U(36 / 20),
        p: {
          marginTop: e(24, 20),
          marginBottom: e(24, 20)
        },
        '[class~="lead"]': {
          fontSize: e(24, 20),
          lineHeight: U(36 / 24),
          marginTop: e(24, 24),
          marginBottom: e(24, 24)
        },
        blockquote: {
          marginTop: e(48, 30),
          marginBottom: e(48, 30),
          paddingInlineStart: e(32, 30)
        },
        h1: {
          fontSize: e(56, 20),
          marginTop: "0",
          marginBottom: e(48, 56),
          lineHeight: U(56 / 56)
        },
        h2: {
          fontSize: e(36, 20),
          marginTop: e(56, 36),
          marginBottom: e(32, 36),
          lineHeight: U(40 / 36)
        },
        h3: {
          fontSize: e(30, 20),
          marginTop: e(48, 30),
          marginBottom: e(20, 30),
          lineHeight: U(40 / 30)
        },
        h4: {
          marginTop: e(36, 20),
          marginBottom: e(12, 20),
          lineHeight: U(32 / 20)
        },
        img: {
          marginTop: e(40, 20),
          marginBottom: e(40, 20)
        },
        picture: {
          marginTop: e(40, 20),
          marginBottom: e(40, 20)
        },
        "picture > img": {
          marginTop: "0",
          marginBottom: "0"
        },
        video: {
          marginTop: e(40, 20),
          marginBottom: e(40, 20)
        },
        kbd: {
          fontSize: e(18, 20),
          borderRadius: me(5),
          paddingTop: e(5, 20),
          paddingInlineEnd: e(8, 20),
          paddingBottom: e(5, 20),
          paddingInlineStart: e(8, 20)
        },
        code: {
          fontSize: e(18, 20)
        },
        "h2 code": {
          fontSize: e(31, 36)
        },
        "h3 code": {
          fontSize: e(27, 30)
        },
        pre: {
          fontSize: e(18, 20),
          lineHeight: U(32 / 18),
          marginTop: e(36, 18),
          marginBottom: e(36, 18),
          borderRadius: me(8),
          paddingTop: e(20, 18),
          paddingInlineEnd: e(24, 18),
          paddingBottom: e(20, 18),
          paddingInlineStart: e(24, 18)
        },
        ol: {
          marginTop: e(24, 20),
          marginBottom: e(24, 20),
          paddingInlineStart: e(32, 20)
        },
        ul: {
          marginTop: e(24, 20),
          marginBottom: e(24, 20),
          paddingInlineStart: e(32, 20)
        },
        li: {
          marginTop: e(12, 20),
          marginBottom: e(12, 20)
        },
        "ol > li": {
          paddingInlineStart: e(8, 20)
        },
        "ul > li": {
          paddingInlineStart: e(8, 20)
        },
        "> ul > li p": {
          marginTop: e(16, 20),
          marginBottom: e(16, 20)
        },
        "> ul > li > p:first-child": {
          marginTop: e(24, 20)
        },
        "> ul > li > p:last-child": {
          marginBottom: e(24, 20)
        },
        "> ol > li > p:first-child": {
          marginTop: e(24, 20)
        },
        "> ol > li > p:last-child": {
          marginBottom: e(24, 20)
        },
        "ul ul, ul ol, ol ul, ol ol": {
          marginTop: e(16, 20),
          marginBottom: e(16, 20)
        },
        dl: {
          marginTop: e(24, 20),
          marginBottom: e(24, 20)
        },
        dt: {
          marginTop: e(24, 20)
        },
        dd: {
          marginTop: e(12, 20),
          paddingInlineStart: e(32, 20)
        },
        hr: {
          marginTop: e(56, 20),
          marginBottom: e(56, 20)
        },
        "hr + *": {
          marginTop: "0"
        },
        "h2 + *": {
          marginTop: "0"
        },
        "h3 + *": {
          marginTop: "0"
        },
        "h4 + *": {
          marginTop: "0"
        },
        table: {
          fontSize: e(18, 20),
          lineHeight: U(28 / 18)
        },
        "thead th": {
          paddingInlineEnd: e(12, 18),
          paddingBottom: e(16, 18),
          paddingInlineStart: e(12, 18)
        },
        "thead th:first-child": {
          paddingInlineStart: "0"
        },
        "thead th:last-child": {
          paddingInlineEnd: "0"
        },
        "tbody td, tfoot td": {
          paddingTop: e(16, 18),
          paddingInlineEnd: e(12, 18),
          paddingBottom: e(16, 18),
          paddingInlineStart: e(12, 18)
        },
        "tbody td:first-child, tfoot td:first-child": {
          paddingInlineStart: "0"
        },
        "tbody td:last-child, tfoot td:last-child": {
          paddingInlineEnd: "0"
        },
        figure: {
          marginTop: e(40, 20),
          marginBottom: e(40, 20)
        },
        "figure > *": {
          marginTop: "0",
          marginBottom: "0"
        },
        figcaption: {
          fontSize: e(18, 20),
          lineHeight: U(28 / 18),
          marginTop: e(18, 18)
        }
      },
      {
        "> :first-child": {
          marginTop: "0"
        },
        "> :last-child": {
          marginBottom: "0"
        }
      }
    ]
  },
  "2xl": {
    css: [
      {
        fontSize: me(24),
        lineHeight: U(40 / 24),
        p: {
          marginTop: e(32, 24),
          marginBottom: e(32, 24)
        },
        '[class~="lead"]': {
          fontSize: e(30, 24),
          lineHeight: U(44 / 30),
          marginTop: e(32, 30),
          marginBottom: e(32, 30)
        },
        blockquote: {
          marginTop: e(64, 36),
          marginBottom: e(64, 36),
          paddingInlineStart: e(40, 36)
        },
        h1: {
          fontSize: e(64, 24),
          marginTop: "0",
          marginBottom: e(56, 64),
          lineHeight: U(64 / 64)
        },
        h2: {
          fontSize: e(48, 24),
          marginTop: e(72, 48),
          marginBottom: e(40, 48),
          lineHeight: U(52 / 48)
        },
        h3: {
          fontSize: e(36, 24),
          marginTop: e(56, 36),
          marginBottom: e(24, 36),
          lineHeight: U(44 / 36)
        },
        h4: {
          marginTop: e(40, 24),
          marginBottom: e(16, 24),
          lineHeight: U(36 / 24)
        },
        img: {
          marginTop: e(48, 24),
          marginBottom: e(48, 24)
        },
        picture: {
          marginTop: e(48, 24),
          marginBottom: e(48, 24)
        },
        "picture > img": {
          marginTop: "0",
          marginBottom: "0"
        },
        video: {
          marginTop: e(48, 24),
          marginBottom: e(48, 24)
        },
        kbd: {
          fontSize: e(20, 24),
          borderRadius: me(6),
          paddingTop: e(6, 24),
          paddingInlineEnd: e(8, 24),
          paddingBottom: e(6, 24),
          paddingInlineStart: e(8, 24)
        },
        code: {
          fontSize: e(20, 24)
        },
        "h2 code": {
          fontSize: e(42, 48)
        },
        "h3 code": {
          fontSize: e(32, 36)
        },
        pre: {
          fontSize: e(20, 24),
          lineHeight: U(36 / 20),
          marginTop: e(40, 20),
          marginBottom: e(40, 20),
          borderRadius: me(8),
          paddingTop: e(24, 20),
          paddingInlineEnd: e(32, 20),
          paddingBottom: e(24, 20),
          paddingInlineStart: e(32, 20)
        },
        ol: {
          marginTop: e(32, 24),
          marginBottom: e(32, 24),
          paddingInlineStart: e(38, 24)
        },
        ul: {
          marginTop: e(32, 24),
          marginBottom: e(32, 24),
          paddingInlineStart: e(38, 24)
        },
        li: {
          marginTop: e(12, 24),
          marginBottom: e(12, 24)
        },
        "ol > li": {
          paddingInlineStart: e(10, 24)
        },
        "ul > li": {
          paddingInlineStart: e(10, 24)
        },
        "> ul > li p": {
          marginTop: e(20, 24),
          marginBottom: e(20, 24)
        },
        "> ul > li > p:first-child": {
          marginTop: e(32, 24)
        },
        "> ul > li > p:last-child": {
          marginBottom: e(32, 24)
        },
        "> ol > li > p:first-child": {
          marginTop: e(32, 24)
        },
        "> ol > li > p:last-child": {
          marginBottom: e(32, 24)
        },
        "ul ul, ul ol, ol ul, ol ol": {
          marginTop: e(16, 24),
          marginBottom: e(16, 24)
        },
        dl: {
          marginTop: e(32, 24),
          marginBottom: e(32, 24)
        },
        dt: {
          marginTop: e(32, 24)
        },
        dd: {
          marginTop: e(12, 24),
          paddingInlineStart: e(38, 24)
        },
        hr: {
          marginTop: e(72, 24),
          marginBottom: e(72, 24)
        },
        "hr + *": {
          marginTop: "0"
        },
        "h2 + *": {
          marginTop: "0"
        },
        "h3 + *": {
          marginTop: "0"
        },
        "h4 + *": {
          marginTop: "0"
        },
        table: {
          fontSize: e(20, 24),
          lineHeight: U(28 / 20)
        },
        "thead th": {
          paddingInlineEnd: e(12, 20),
          paddingBottom: e(16, 20),
          paddingInlineStart: e(12, 20)
        },
        "thead th:first-child": {
          paddingInlineStart: "0"
        },
        "thead th:last-child": {
          paddingInlineEnd: "0"
        },
        "tbody td, tfoot td": {
          paddingTop: e(16, 20),
          paddingInlineEnd: e(12, 20),
          paddingBottom: e(16, 20),
          paddingInlineStart: e(12, 20)
        },
        "tbody td:first-child, tfoot td:first-child": {
          paddingInlineStart: "0"
        },
        "tbody td:last-child, tfoot td:last-child": {
          paddingInlineEnd: "0"
        },
        figure: {
          marginTop: e(48, 24),
          marginBottom: e(48, 24)
        },
        "figure > *": {
          marginTop: "0",
          marginBottom: "0"
        },
        figcaption: {
          fontSize: e(20, 24),
          lineHeight: U(32 / 20),
          marginTop: e(20, 20)
        }
      },
      {
        "> :first-child": {
          marginTop: "0"
        },
        "> :last-child": {
          marginBottom: "0"
        }
      }
    ]
  },
  // Gray color themes
  slate: {
    css: {
      "--tw-prose-body": s.slate[700],
      "--tw-prose-headings": s.slate[900],
      "--tw-prose-lead": s.slate[600],
      "--tw-prose-links": s.slate[900],
      "--tw-prose-bold": s.slate[900],
      "--tw-prose-counters": s.slate[500],
      "--tw-prose-bullets": s.slate[300],
      "--tw-prose-hr": s.slate[200],
      "--tw-prose-quotes": s.slate[900],
      "--tw-prose-quote-borders": s.slate[200],
      "--tw-prose-captions": s.slate[500],
      "--tw-prose-kbd": s.slate[900],
      "--tw-prose-kbd-shadows": Ae(s.slate[900]),
      "--tw-prose-code": s.slate[900],
      "--tw-prose-pre-code": s.slate[200],
      "--tw-prose-pre-bg": s.slate[800],
      "--tw-prose-th-borders": s.slate[300],
      "--tw-prose-td-borders": s.slate[200],
      "--tw-prose-invert-body": s.slate[300],
      "--tw-prose-invert-headings": s.white,
      "--tw-prose-invert-lead": s.slate[400],
      "--tw-prose-invert-links": s.white,
      "--tw-prose-invert-bold": s.white,
      "--tw-prose-invert-counters": s.slate[400],
      "--tw-prose-invert-bullets": s.slate[600],
      "--tw-prose-invert-hr": s.slate[700],
      "--tw-prose-invert-quotes": s.slate[100],
      "--tw-prose-invert-quote-borders": s.slate[700],
      "--tw-prose-invert-captions": s.slate[400],
      "--tw-prose-invert-kbd": s.white,
      "--tw-prose-invert-kbd-shadows": Ae(s.white),
      "--tw-prose-invert-code": s.white,
      "--tw-prose-invert-pre-code": s.slate[300],
      "--tw-prose-invert-pre-bg": "rgb(0 0 0 / 50%)",
      "--tw-prose-invert-th-borders": s.slate[600],
      "--tw-prose-invert-td-borders": s.slate[700]
    }
  },
  gray: {
    css: {
      "--tw-prose-body": s.gray[700],
      "--tw-prose-headings": s.gray[900],
      "--tw-prose-lead": s.gray[600],
      "--tw-prose-links": s.gray[900],
      "--tw-prose-bold": s.gray[900],
      "--tw-prose-counters": s.gray[500],
      "--tw-prose-bullets": s.gray[300],
      "--tw-prose-hr": s.gray[200],
      "--tw-prose-quotes": s.gray[900],
      "--tw-prose-quote-borders": s.gray[200],
      "--tw-prose-captions": s.gray[500],
      "--tw-prose-kbd": s.gray[900],
      "--tw-prose-kbd-shadows": Ae(s.gray[900]),
      "--tw-prose-code": s.gray[900],
      "--tw-prose-pre-code": s.gray[200],
      "--tw-prose-pre-bg": s.gray[800],
      "--tw-prose-th-borders": s.gray[300],
      "--tw-prose-td-borders": s.gray[200],
      "--tw-prose-invert-body": s.gray[300],
      "--tw-prose-invert-headings": s.white,
      "--tw-prose-invert-lead": s.gray[400],
      "--tw-prose-invert-links": s.white,
      "--tw-prose-invert-bold": s.white,
      "--tw-prose-invert-counters": s.gray[400],
      "--tw-prose-invert-bullets": s.gray[600],
      "--tw-prose-invert-hr": s.gray[700],
      "--tw-prose-invert-quotes": s.gray[100],
      "--tw-prose-invert-quote-borders": s.gray[700],
      "--tw-prose-invert-captions": s.gray[400],
      "--tw-prose-invert-kbd": s.white,
      "--tw-prose-invert-kbd-shadows": Ae(s.white),
      "--tw-prose-invert-code": s.white,
      "--tw-prose-invert-pre-code": s.gray[300],
      "--tw-prose-invert-pre-bg": "rgb(0 0 0 / 50%)",
      "--tw-prose-invert-th-borders": s.gray[600],
      "--tw-prose-invert-td-borders": s.gray[700]
    }
  },
  zinc: {
    css: {
      "--tw-prose-body": s.zinc[700],
      "--tw-prose-headings": s.zinc[900],
      "--tw-prose-lead": s.zinc[600],
      "--tw-prose-links": s.zinc[900],
      "--tw-prose-bold": s.zinc[900],
      "--tw-prose-counters": s.zinc[500],
      "--tw-prose-bullets": s.zinc[300],
      "--tw-prose-hr": s.zinc[200],
      "--tw-prose-quotes": s.zinc[900],
      "--tw-prose-quote-borders": s.zinc[200],
      "--tw-prose-captions": s.zinc[500],
      "--tw-prose-kbd": s.zinc[900],
      "--tw-prose-kbd-shadows": Ae(s.zinc[900]),
      "--tw-prose-code": s.zinc[900],
      "--tw-prose-pre-code": s.zinc[200],
      "--tw-prose-pre-bg": s.zinc[800],
      "--tw-prose-th-borders": s.zinc[300],
      "--tw-prose-td-borders": s.zinc[200],
      "--tw-prose-invert-body": s.zinc[300],
      "--tw-prose-invert-headings": s.white,
      "--tw-prose-invert-lead": s.zinc[400],
      "--tw-prose-invert-links": s.white,
      "--tw-prose-invert-bold": s.white,
      "--tw-prose-invert-counters": s.zinc[400],
      "--tw-prose-invert-bullets": s.zinc[600],
      "--tw-prose-invert-hr": s.zinc[700],
      "--tw-prose-invert-quotes": s.zinc[100],
      "--tw-prose-invert-quote-borders": s.zinc[700],
      "--tw-prose-invert-captions": s.zinc[400],
      "--tw-prose-invert-kbd": s.white,
      "--tw-prose-invert-kbd-shadows": Ae(s.white),
      "--tw-prose-invert-code": s.white,
      "--tw-prose-invert-pre-code": s.zinc[300],
      "--tw-prose-invert-pre-bg": "rgb(0 0 0 / 50%)",
      "--tw-prose-invert-th-borders": s.zinc[600],
      "--tw-prose-invert-td-borders": s.zinc[700]
    }
  },
  neutral: {
    css: {
      "--tw-prose-body": s.neutral[700],
      "--tw-prose-headings": s.neutral[900],
      "--tw-prose-lead": s.neutral[600],
      "--tw-prose-links": s.neutral[900],
      "--tw-prose-bold": s.neutral[900],
      "--tw-prose-counters": s.neutral[500],
      "--tw-prose-bullets": s.neutral[300],
      "--tw-prose-hr": s.neutral[200],
      "--tw-prose-quotes": s.neutral[900],
      "--tw-prose-quote-borders": s.neutral[200],
      "--tw-prose-captions": s.neutral[500],
      "--tw-prose-kbd": s.neutral[900],
      "--tw-prose-kbd-shadows": Ae(s.neutral[900]),
      "--tw-prose-code": s.neutral[900],
      "--tw-prose-pre-code": s.neutral[200],
      "--tw-prose-pre-bg": s.neutral[800],
      "--tw-prose-th-borders": s.neutral[300],
      "--tw-prose-td-borders": s.neutral[200],
      "--tw-prose-invert-body": s.neutral[300],
      "--tw-prose-invert-headings": s.white,
      "--tw-prose-invert-lead": s.neutral[400],
      "--tw-prose-invert-links": s.white,
      "--tw-prose-invert-bold": s.white,
      "--tw-prose-invert-counters": s.neutral[400],
      "--tw-prose-invert-bullets": s.neutral[600],
      "--tw-prose-invert-hr": s.neutral[700],
      "--tw-prose-invert-quotes": s.neutral[100],
      "--tw-prose-invert-quote-borders": s.neutral[700],
      "--tw-prose-invert-captions": s.neutral[400],
      "--tw-prose-invert-kbd": s.white,
      "--tw-prose-invert-kbd-shadows": Ae(s.white),
      "--tw-prose-invert-code": s.white,
      "--tw-prose-invert-pre-code": s.neutral[300],
      "--tw-prose-invert-pre-bg": "rgb(0 0 0 / 50%)",
      "--tw-prose-invert-th-borders": s.neutral[600],
      "--tw-prose-invert-td-borders": s.neutral[700]
    }
  },
  stone: {
    css: {
      "--tw-prose-body": s.stone[700],
      "--tw-prose-headings": s.stone[900],
      "--tw-prose-lead": s.stone[600],
      "--tw-prose-links": s.stone[900],
      "--tw-prose-bold": s.stone[900],
      "--tw-prose-counters": s.stone[500],
      "--tw-prose-bullets": s.stone[300],
      "--tw-prose-hr": s.stone[200],
      "--tw-prose-quotes": s.stone[900],
      "--tw-prose-quote-borders": s.stone[200],
      "--tw-prose-captions": s.stone[500],
      "--tw-prose-kbd": s.stone[900],
      "--tw-prose-kbd-shadows": Ae(s.stone[900]),
      "--tw-prose-code": s.stone[900],
      "--tw-prose-pre-code": s.stone[200],
      "--tw-prose-pre-bg": s.stone[800],
      "--tw-prose-th-borders": s.stone[300],
      "--tw-prose-td-borders": s.stone[200],
      "--tw-prose-invert-body": s.stone[300],
      "--tw-prose-invert-headings": s.white,
      "--tw-prose-invert-lead": s.stone[400],
      "--tw-prose-invert-links": s.white,
      "--tw-prose-invert-bold": s.white,
      "--tw-prose-invert-counters": s.stone[400],
      "--tw-prose-invert-bullets": s.stone[600],
      "--tw-prose-invert-hr": s.stone[700],
      "--tw-prose-invert-quotes": s.stone[100],
      "--tw-prose-invert-quote-borders": s.stone[700],
      "--tw-prose-invert-captions": s.stone[400],
      "--tw-prose-invert-kbd": s.white,
      "--tw-prose-invert-kbd-shadows": Ae(s.white),
      "--tw-prose-invert-code": s.white,
      "--tw-prose-invert-pre-code": s.stone[300],
      "--tw-prose-invert-pre-bg": "rgb(0 0 0 / 50%)",
      "--tw-prose-invert-th-borders": s.stone[600],
      "--tw-prose-invert-td-borders": s.stone[700]
    }
  },
  // Link-only themes (for backward compatibility)
  red: {
    css: {
      "--tw-prose-links": s.red[600],
      "--tw-prose-invert-links": s.red[500]
    }
  },
  orange: {
    css: {
      "--tw-prose-links": s.orange[600],
      "--tw-prose-invert-links": s.orange[500]
    }
  },
  amber: {
    css: {
      "--tw-prose-links": s.amber[600],
      "--tw-prose-invert-links": s.amber[500]
    }
  },
  yellow: {
    css: {
      "--tw-prose-links": s.yellow[600],
      "--tw-prose-invert-links": s.yellow[500]
    }
  },
  lime: {
    css: {
      "--tw-prose-links": s.lime[600],
      "--tw-prose-invert-links": s.lime[500]
    }
  },
  green: {
    css: {
      "--tw-prose-links": s.green[600],
      "--tw-prose-invert-links": s.green[500]
    }
  },
  emerald: {
    css: {
      "--tw-prose-links": s.emerald[600],
      "--tw-prose-invert-links": s.emerald[500]
    }
  },
  teal: {
    css: {
      "--tw-prose-links": s.teal[600],
      "--tw-prose-invert-links": s.teal[500]
    }
  },
  cyan: {
    css: {
      "--tw-prose-links": s.cyan[600],
      "--tw-prose-invert-links": s.cyan[500]
    }
  },
  sky: {
    css: {
      "--tw-prose-links": s.sky[600],
      "--tw-prose-invert-links": s.sky[500]
    }
  },
  blue: {
    css: {
      "--tw-prose-links": s.blue[600],
      "--tw-prose-invert-links": s.blue[500]
    }
  },
  indigo: {
    css: {
      "--tw-prose-links": s.indigo[600],
      "--tw-prose-invert-links": s.indigo[500]
    }
  },
  violet: {
    css: {
      "--tw-prose-links": s.violet[600],
      "--tw-prose-invert-links": s.violet[500]
    }
  },
  purple: {
    css: {
      "--tw-prose-links": s.purple[600],
      "--tw-prose-invert-links": s.purple[500]
    }
  },
  fuchsia: {
    css: {
      "--tw-prose-links": s.fuchsia[600],
      "--tw-prose-invert-links": s.fuchsia[500]
    }
  },
  pink: {
    css: {
      "--tw-prose-links": s.pink[600],
      "--tw-prose-invert-links": s.pink[500]
    }
  },
  rose: {
    css: {
      "--tw-prose-links": s.rose[600],
      "--tw-prose-invert-links": s.rose[500]
    }
  },
  // Invert (for dark mode)
  invert: {
    css: {
      "--tw-prose-body": "var(--tw-prose-invert-body)",
      "--tw-prose-headings": "var(--tw-prose-invert-headings)",
      "--tw-prose-lead": "var(--tw-prose-invert-lead)",
      "--tw-prose-links": "var(--tw-prose-invert-links)",
      "--tw-prose-bold": "var(--tw-prose-invert-bold)",
      "--tw-prose-counters": "var(--tw-prose-invert-counters)",
      "--tw-prose-bullets": "var(--tw-prose-invert-bullets)",
      "--tw-prose-hr": "var(--tw-prose-invert-hr)",
      "--tw-prose-quotes": "var(--tw-prose-invert-quotes)",
      "--tw-prose-quote-borders": "var(--tw-prose-invert-quote-borders)",
      "--tw-prose-captions": "var(--tw-prose-invert-captions)",
      "--tw-prose-kbd": "var(--tw-prose-invert-kbd)",
      "--tw-prose-kbd-shadows": "var(--tw-prose-invert-kbd-shadows)",
      "--tw-prose-code": "var(--tw-prose-invert-code)",
      "--tw-prose-pre-code": "var(--tw-prose-invert-pre-code)",
      "--tw-prose-pre-bg": "var(--tw-prose-invert-pre-bg)",
      "--tw-prose-th-borders": "var(--tw-prose-invert-th-borders)",
      "--tw-prose-td-borders": "var(--tw-prose-invert-td-borders)"
    }
  }
};
var Jn = {
  DEFAULT: {
    css: [
      {
        color: "var(--tw-prose-body)",
        maxWidth: "65ch",
        p: {},
        // Required to maintain correct order when merging
        '[class~="lead"]': {
          color: "var(--tw-prose-lead)"
        },
        a: {
          color: "var(--tw-prose-links)",
          textDecoration: "underline",
          fontWeight: "500"
        },
        strong: {
          color: "var(--tw-prose-bold)",
          fontWeight: "600"
        },
        "a strong": {
          color: "inherit"
        },
        "blockquote strong": {
          color: "inherit"
        },
        "thead th strong": {
          color: "inherit"
        },
        ol: {
          listStyleType: "decimal"
        },
        'ol[type="A"]': {
          listStyleType: "upper-alpha"
        },
        'ol[type="a"]': {
          listStyleType: "lower-alpha"
        },
        'ol[type="A" s]': {
          listStyleType: "upper-alpha"
        },
        'ol[type="a" s]': {
          listStyleType: "lower-alpha"
        },
        'ol[type="I"]': {
          listStyleType: "upper-roman"
        },
        'ol[type="i"]': {
          listStyleType: "lower-roman"
        },
        'ol[type="I" s]': {
          listStyleType: "upper-roman"
        },
        'ol[type="i" s]': {
          listStyleType: "lower-roman"
        },
        'ol[type="1"]': {
          listStyleType: "decimal"
        },
        ul: {
          listStyleType: "disc"
        },
        "ol > li::marker": {
          fontWeight: "400",
          color: "var(--tw-prose-counters)"
        },
        "ul > li::marker": {
          color: "var(--tw-prose-bullets)"
        },
        dt: {
          color: "var(--tw-prose-headings)",
          fontWeight: "600"
        },
        hr: {
          borderColor: "var(--tw-prose-hr)",
          borderTopWidth: 1
        },
        blockquote: {
          fontWeight: "500",
          fontStyle: "italic",
          color: "var(--tw-prose-quotes)",
          borderInlineStartWidth: "0.25rem",
          borderInlineStartColor: "var(--tw-prose-quote-borders)",
          quotes: '"\\201C""\\201D""\\2018""\\2019"'
        },
        "blockquote p:first-of-type::before": {
          content: "open-quote"
        },
        "blockquote p:last-of-type::after": {
          content: "close-quote"
        },
        h1: {
          color: "var(--tw-prose-headings)",
          fontWeight: "800"
        },
        "h1 strong": {
          fontWeight: "900",
          color: "inherit"
        },
        h2: {
          color: "var(--tw-prose-headings)",
          fontWeight: "700"
        },
        "h2 strong": {
          fontWeight: "800",
          color: "inherit"
        },
        h3: {
          color: "var(--tw-prose-headings)",
          fontWeight: "600"
        },
        "h3 strong": {
          fontWeight: "700",
          color: "inherit"
        },
        h4: {
          color: "var(--tw-prose-headings)",
          fontWeight: "600"
        },
        "h4 strong": {
          fontWeight: "700",
          color: "inherit"
        },
        img: {},
        // Required to maintain correct order when merging
        picture: {
          display: "block"
        },
        video: {},
        // Required to maintain correct order when merging
        kbd: {
          fontWeight: "500",
          fontFamily: "inherit",
          color: "var(--tw-prose-kbd)",
          boxShadow: "0 0 0 1px rgb(var(--tw-prose-kbd-shadows) / 10%), 0 3px 0 rgb(var(--tw-prose-kbd-shadows) / 10%)"
        },
        code: {
          color: "var(--tw-prose-code)",
          fontWeight: "600"
        },
        "code::before": {
          content: '"`"'
        },
        "code::after": {
          content: '"`"'
        },
        "a code": {
          color: "inherit"
        },
        "h1 code": {
          color: "inherit"
        },
        "h2 code": {
          color: "inherit"
        },
        "h3 code": {
          color: "inherit"
        },
        "h4 code": {
          color: "inherit"
        },
        "blockquote code": {
          color: "inherit"
        },
        "thead th code": {
          color: "inherit"
        },
        pre: {
          color: "var(--tw-prose-pre-code)",
          backgroundColor: "var(--tw-prose-pre-bg)",
          overflowX: "auto",
          fontWeight: "400"
        },
        "pre code": {
          backgroundColor: "transparent",
          borderWidth: "0",
          borderRadius: "0",
          padding: "0",
          fontWeight: "inherit",
          color: "inherit",
          fontSize: "inherit",
          fontFamily: "inherit",
          lineHeight: "inherit"
        },
        "pre code::before": {
          content: "none"
        },
        "pre code::after": {
          content: "none"
        },
        table: {
          width: "100%",
          tableLayout: "auto",
          textAlign: "start",
          marginTop: e(32, 16),
          marginBottom: e(32, 16)
        },
        thead: {
          borderBottomWidth: "1px",
          borderBottomColor: "var(--tw-prose-th-borders)"
        },
        "thead th": {
          color: "var(--tw-prose-headings)",
          fontWeight: "600",
          verticalAlign: "bottom"
        },
        "tbody tr": {
          borderBottomWidth: "1px",
          borderBottomColor: "var(--tw-prose-td-borders)"
        },
        "tbody tr:last-child": {
          borderBottomWidth: "0"
        },
        "tbody td": {
          verticalAlign: "baseline"
        },
        tfoot: {
          borderTopWidth: "1px",
          borderTopColor: "var(--tw-prose-th-borders)"
        },
        "tfoot td": {
          verticalAlign: "top"
        },
        "figure > *": {},
        // Required to maintain correct order when merging
        figcaption: {
          color: "var(--tw-prose-captions)"
        }
      },
      vt.gray.css,
      ...vt.base.css
    ]
  },
  ...vt
}, Xn = "[object Object]";
function Zn(a) {
  var t = !1;
  if (a != null && typeof a.toString != "function")
    try {
      t = !!(a + "");
    } catch {
    }
  return t;
}
function ei(a, t) {
  return function(b) {
    return a(t(b));
  };
}
var ti = Function.prototype, gr = Object.prototype, vr = ti.toString, ri = gr.hasOwnProperty, ni = vr.call(Object), ii = gr.toString, oi = ei(Object.getPrototypeOf, Object);
function ai(a) {
  return !!a && typeof a == "object";
}
function si(a) {
  if (!ai(a) || ii.call(a) != Xn || Zn(a))
    return !1;
  var t = oi(a);
  if (t === null)
    return !0;
  var b = ri.call(t, "constructor") && t.constructor;
  return typeof b == "function" && b instanceof b && vr.call(b) == ni;
}
var ui = si, wt = { exports: {} }, bt = { exports: {} }, _t = { exports: {} }, Tt = { exports: {} }, yt = { exports: {} }, St = { exports: {} }, ye = {}, Et = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = v;
  function b(y) {
    for (var f = y.toLowerCase(), g = "", n = !1, o = 0; o < 6 && f[o] !== void 0; o++) {
      var l = f.charCodeAt(o), c = l >= 97 && l <= 102 || l >= 48 && l <= 57;
      if (n = l === 32, !c)
        break;
      g += f[o];
    }
    if (g.length !== 0) {
      var w = parseInt(g, 16), B = w >= 55296 && w <= 57343;
      return B || w === 0 || w > 1114111 ? ["", g.length + (n ? 1 : 0)] : [String.fromCodePoint(w), g.length + (n ? 1 : 0)];
    }
  }
  var p = /\\/;
  function v(y) {
    var f = p.test(y);
    if (!f)
      return y;
    for (var g = "", n = 0; n < y.length; n++) {
      if (y[n] === "\\") {
        var o = b(y.slice(n + 1, n + 7));
        if (o !== void 0) {
          g += o[0], n += o[1];
          continue;
        }
        if (y[n + 1] === "\\") {
          g += "\\", n++;
          continue;
        }
        y.length === n + 1 && (g += y[n]);
        continue;
      }
      g += y[n];
    }
    return g;
  }
  a.exports = t.default;
})(Et, Et.exports);
var mr = Et.exports, Ot = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = b;
  function b(p) {
    for (var v = arguments.length, y = new Array(v > 1 ? v - 1 : 0), f = 1; f < v; f++)
      y[f - 1] = arguments[f];
    for (; y.length > 0; ) {
      var g = y.shift();
      if (!p[g])
        return;
      p = p[g];
    }
    return p;
  }
  a.exports = t.default;
})(Ot, Ot.exports);
var li = Ot.exports, It = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = b;
  function b(p) {
    for (var v = arguments.length, y = new Array(v > 1 ? v - 1 : 0), f = 1; f < v; f++)
      y[f - 1] = arguments[f];
    for (; y.length > 0; ) {
      var g = y.shift();
      p[g] || (p[g] = {}), p = p[g];
    }
  }
  a.exports = t.default;
})(It, It.exports);
var ci = It.exports, kt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = b;
  function b(p) {
    for (var v = "", y = p.indexOf("/*"), f = 0; y >= 0; ) {
      v = v + p.slice(f, y);
      var g = p.indexOf("*/", y + 2);
      if (g < 0)
        return v;
      f = g + 2, y = p.indexOf("/*", f);
    }
    return v = v + p.slice(f), v;
  }
  a.exports = t.default;
})(kt, kt.exports);
var fi = kt.exports;
ye.__esModule = !0;
ye.stripComments = ye.ensureObject = ye.getProp = ye.unesc = void 0;
var pi = tt(mr);
ye.unesc = pi.default;
var di = tt(li);
ye.getProp = di.default;
var hi = tt(ci);
ye.ensureObject = hi.default;
var gi = tt(fi);
ye.stripComments = gi.default;
function tt(a) {
  return a && a.__esModule ? a : { default: a };
}
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = ye;
  function p(g, n) {
    for (var o = 0; o < n.length; o++) {
      var l = n[o];
      l.enumerable = l.enumerable || !1, l.configurable = !0, "value" in l && (l.writable = !0), Object.defineProperty(g, l.key, l);
    }
  }
  function v(g, n, o) {
    return n && p(g.prototype, n), g;
  }
  var y = function g(n, o) {
    if (typeof n != "object" || n === null)
      return n;
    var l = new n.constructor();
    for (var c in n)
      if (n.hasOwnProperty(c)) {
        var w = n[c], B = typeof w;
        c === "parent" && B === "object" ? o && (l[c] = o) : w instanceof Array ? l[c] = w.map(function(F) {
          return g(F, l);
        }) : l[c] = g(w, l);
      }
    return l;
  }, f = /* @__PURE__ */ function() {
    function g(o) {
      o === void 0 && (o = {}), Object.assign(this, o), this.spaces = this.spaces || {}, this.spaces.before = this.spaces.before || "", this.spaces.after = this.spaces.after || "";
    }
    var n = g.prototype;
    return n.remove = function() {
      return this.parent && this.parent.removeChild(this), this.parent = void 0, this;
    }, n.replaceWith = function() {
      if (this.parent) {
        for (var l in arguments)
          this.parent.insertBefore(this, arguments[l]);
        this.remove();
      }
      return this;
    }, n.next = function() {
      return this.parent.at(this.parent.index(this) + 1);
    }, n.prev = function() {
      return this.parent.at(this.parent.index(this) - 1);
    }, n.clone = function(l) {
      l === void 0 && (l = {});
      var c = y(this);
      for (var w in l)
        c[w] = l[w];
      return c;
    }, n.appendToPropertyAndEscape = function(l, c, w) {
      this.raws || (this.raws = {});
      var B = this[l], F = this.raws[l];
      this[l] = B + c, F || w !== c ? this.raws[l] = (F || B) + w : delete this.raws[l];
    }, n.setPropertyAndEscape = function(l, c, w) {
      this.raws || (this.raws = {}), this[l] = c, this.raws[l] = w;
    }, n.setPropertyWithoutEscape = function(l, c) {
      this[l] = c, this.raws && delete this.raws[l];
    }, n.isAtPosition = function(l, c) {
      if (this.source && this.source.start && this.source.end)
        return !(this.source.start.line > l || this.source.end.line < l || this.source.start.line === l && this.source.start.column > c || this.source.end.line === l && this.source.end.column < c);
    }, n.stringifyProperty = function(l) {
      return this.raws && this.raws[l] || this[l];
    }, n.valueToString = function() {
      return String(this.stringifyProperty("value"));
    }, n.toString = function() {
      return [this.rawSpaceBefore, this.valueToString(), this.rawSpaceAfter].join("");
    }, v(g, [{
      key: "rawSpaceBefore",
      get: function() {
        var l = this.raws && this.raws.spaces && this.raws.spaces.before;
        return l === void 0 && (l = this.spaces && this.spaces.before), l || "";
      },
      set: function(l) {
        (0, b.ensureObject)(this, "raws", "spaces"), this.raws.spaces.before = l;
      }
    }, {
      key: "rawSpaceAfter",
      get: function() {
        var l = this.raws && this.raws.spaces && this.raws.spaces.after;
        return l === void 0 && (l = this.spaces.after), l || "";
      },
      set: function(l) {
        (0, b.ensureObject)(this, "raws", "spaces"), this.raws.spaces.after = l;
      }
    }]), g;
  }();
  t.default = f, a.exports = t.default;
})(St, St.exports);
var Le = St.exports, G = {};
G.__esModule = !0;
G.UNIVERSAL = G.ATTRIBUTE = G.CLASS = G.COMBINATOR = G.COMMENT = G.ID = G.NESTING = G.PSEUDO = G.ROOT = G.SELECTOR = G.STRING = G.TAG = void 0;
var vi = "tag";
G.TAG = vi;
var mi = "string";
G.STRING = mi;
var wi = "selector";
G.SELECTOR = wi;
var bi = "root";
G.ROOT = bi;
var _i = "pseudo";
G.PSEUDO = _i;
var Ti = "nesting";
G.NESTING = Ti;
var yi = "id";
G.ID = yi;
var Si = "comment";
G.COMMENT = Si;
var Ei = "combinator";
G.COMBINATOR = Ei;
var Oi = "class";
G.CLASS = Oi;
var Ii = "attribute";
G.ATTRIBUTE = Ii;
var ki = "universal";
G.UNIVERSAL = ki;
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = f(Le), p = y(G);
  function v() {
    if (typeof WeakMap != "function")
      return null;
    var i = /* @__PURE__ */ new WeakMap();
    return v = function() {
      return i;
    }, i;
  }
  function y(i) {
    if (i && i.__esModule)
      return i;
    if (i === null || typeof i != "object" && typeof i != "function")
      return { default: i };
    var d = v();
    if (d && d.has(i))
      return d.get(i);
    var S = {}, O = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var E in i)
      if (Object.prototype.hasOwnProperty.call(i, E)) {
        var k = O ? Object.getOwnPropertyDescriptor(i, E) : null;
        k && (k.get || k.set) ? Object.defineProperty(S, E, k) : S[E] = i[E];
      }
    return S.default = i, d && d.set(i, S), S;
  }
  function f(i) {
    return i && i.__esModule ? i : { default: i };
  }
  function g(i, d) {
    var S;
    if (typeof Symbol > "u" || i[Symbol.iterator] == null) {
      if (Array.isArray(i) || (S = n(i)) || d) {
        S && (i = S);
        var O = 0;
        return function() {
          return O >= i.length ? { done: !0 } : { done: !1, value: i[O++] };
        };
      }
      throw new TypeError(`Invalid attempt to iterate non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
    }
    return S = i[Symbol.iterator](), S.next.bind(S);
  }
  function n(i, d) {
    if (i) {
      if (typeof i == "string")
        return o(i, d);
      var S = Object.prototype.toString.call(i).slice(8, -1);
      if (S === "Object" && i.constructor && (S = i.constructor.name), S === "Map" || S === "Set")
        return Array.from(i);
      if (S === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(S))
        return o(i, d);
    }
  }
  function o(i, d) {
    (d == null || d > i.length) && (d = i.length);
    for (var S = 0, O = new Array(d); S < d; S++)
      O[S] = i[S];
    return O;
  }
  function l(i, d) {
    for (var S = 0; S < d.length; S++) {
      var O = d[S];
      O.enumerable = O.enumerable || !1, O.configurable = !0, "value" in O && (O.writable = !0), Object.defineProperty(i, O.key, O);
    }
  }
  function c(i, d, S) {
    return d && l(i.prototype, d), i;
  }
  function w(i, d) {
    i.prototype = Object.create(d.prototype), i.prototype.constructor = i, B(i, d);
  }
  function B(i, d) {
    return B = Object.setPrototypeOf || function(O, E) {
      return O.__proto__ = E, O;
    }, B(i, d);
  }
  var F = /* @__PURE__ */ function(i) {
    w(d, i);
    function d(O) {
      var E;
      return E = i.call(this, O) || this, E.nodes || (E.nodes = []), E;
    }
    var S = d.prototype;
    return S.append = function(E) {
      return E.parent = this, this.nodes.push(E), this;
    }, S.prepend = function(E) {
      return E.parent = this, this.nodes.unshift(E), this;
    }, S.at = function(E) {
      return this.nodes[E];
    }, S.index = function(E) {
      return typeof E == "number" ? E : this.nodes.indexOf(E);
    }, S.removeChild = function(E) {
      E = this.index(E), this.at(E).parent = void 0, this.nodes.splice(E, 1);
      var k;
      for (var T in this.indexes)
        k = this.indexes[T], k >= E && (this.indexes[T] = k - 1);
      return this;
    }, S.removeAll = function() {
      for (var E = g(this.nodes), k; !(k = E()).done; ) {
        var T = k.value;
        T.parent = void 0;
      }
      return this.nodes = [], this;
    }, S.empty = function() {
      return this.removeAll();
    }, S.insertAfter = function(E, k) {
      k.parent = this;
      var T = this.index(E);
      this.nodes.splice(T + 1, 0, k), k.parent = this;
      var x;
      for (var N in this.indexes)
        x = this.indexes[N], T <= x && (this.indexes[N] = x + 1);
      return this;
    }, S.insertBefore = function(E, k) {
      k.parent = this;
      var T = this.index(E);
      this.nodes.splice(T, 0, k), k.parent = this;
      var x;
      for (var N in this.indexes)
        x = this.indexes[N], x <= T && (this.indexes[N] = x + 1);
      return this;
    }, S._findChildAtPosition = function(E, k) {
      var T = void 0;
      return this.each(function(x) {
        if (x.atPosition) {
          var N = x.atPosition(E, k);
          if (N)
            return T = N, !1;
        } else if (x.isAtPosition(E, k))
          return T = x, !1;
      }), T;
    }, S.atPosition = function(E, k) {
      if (this.isAtPosition(E, k))
        return this._findChildAtPosition(E, k) || this;
    }, S._inferEndPosition = function() {
      this.last && this.last.source && this.last.source.end && (this.source = this.source || {}, this.source.end = this.source.end || {}, Object.assign(this.source.end, this.last.source.end));
    }, S.each = function(E) {
      this.lastEach || (this.lastEach = 0), this.indexes || (this.indexes = {}), this.lastEach++;
      var k = this.lastEach;
      if (this.indexes[k] = 0, !!this.length) {
        for (var T, x; this.indexes[k] < this.length && (T = this.indexes[k], x = E(this.at(T), T), x !== !1); )
          this.indexes[k] += 1;
        if (delete this.indexes[k], x === !1)
          return !1;
      }
    }, S.walk = function(E) {
      return this.each(function(k, T) {
        var x = E(k, T);
        if (x !== !1 && k.length && (x = k.walk(E)), x === !1)
          return !1;
      });
    }, S.walkAttributes = function(E) {
      var k = this;
      return this.walk(function(T) {
        if (T.type === p.ATTRIBUTE)
          return E.call(k, T);
      });
    }, S.walkClasses = function(E) {
      var k = this;
      return this.walk(function(T) {
        if (T.type === p.CLASS)
          return E.call(k, T);
      });
    }, S.walkCombinators = function(E) {
      var k = this;
      return this.walk(function(T) {
        if (T.type === p.COMBINATOR)
          return E.call(k, T);
      });
    }, S.walkComments = function(E) {
      var k = this;
      return this.walk(function(T) {
        if (T.type === p.COMMENT)
          return E.call(k, T);
      });
    }, S.walkIds = function(E) {
      var k = this;
      return this.walk(function(T) {
        if (T.type === p.ID)
          return E.call(k, T);
      });
    }, S.walkNesting = function(E) {
      var k = this;
      return this.walk(function(T) {
        if (T.type === p.NESTING)
          return E.call(k, T);
      });
    }, S.walkPseudos = function(E) {
      var k = this;
      return this.walk(function(T) {
        if (T.type === p.PSEUDO)
          return E.call(k, T);
      });
    }, S.walkTags = function(E) {
      var k = this;
      return this.walk(function(T) {
        if (T.type === p.TAG)
          return E.call(k, T);
      });
    }, S.walkUniversals = function(E) {
      var k = this;
      return this.walk(function(T) {
        if (T.type === p.UNIVERSAL)
          return E.call(k, T);
      });
    }, S.split = function(E) {
      var k = this, T = [];
      return this.reduce(function(x, N, L) {
        var A = E.call(k, N);
        return T.push(N), A ? (x.push(T), T = []) : L === k.length - 1 && x.push(T), x;
      }, []);
    }, S.map = function(E) {
      return this.nodes.map(E);
    }, S.reduce = function(E, k) {
      return this.nodes.reduce(E, k);
    }, S.every = function(E) {
      return this.nodes.every(E);
    }, S.some = function(E) {
      return this.nodes.some(E);
    }, S.filter = function(E) {
      return this.nodes.filter(E);
    }, S.sort = function(E) {
      return this.nodes.sort(E);
    }, S.toString = function() {
      return this.map(String).join("");
    }, c(d, [{
      key: "first",
      get: function() {
        return this.at(0);
      }
    }, {
      key: "last",
      get: function() {
        return this.at(this.length - 1);
      }
    }, {
      key: "length",
      get: function() {
        return this.nodes.length;
      }
    }]), d;
  }(b.default);
  t.default = F, a.exports = t.default;
})(yt, yt.exports);
var Ut = yt.exports;
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = v(Ut), p = G;
  function v(l) {
    return l && l.__esModule ? l : { default: l };
  }
  function y(l, c) {
    for (var w = 0; w < c.length; w++) {
      var B = c[w];
      B.enumerable = B.enumerable || !1, B.configurable = !0, "value" in B && (B.writable = !0), Object.defineProperty(l, B.key, B);
    }
  }
  function f(l, c, w) {
    return c && y(l.prototype, c), l;
  }
  function g(l, c) {
    l.prototype = Object.create(c.prototype), l.prototype.constructor = l, n(l, c);
  }
  function n(l, c) {
    return n = Object.setPrototypeOf || function(B, F) {
      return B.__proto__ = F, B;
    }, n(l, c);
  }
  var o = /* @__PURE__ */ function(l) {
    g(c, l);
    function c(B) {
      var F;
      return F = l.call(this, B) || this, F.type = p.ROOT, F;
    }
    var w = c.prototype;
    return w.toString = function() {
      var F = this.reduce(function(i, d) {
        return i.push(String(d)), i;
      }, []).join(",");
      return this.trailingComma ? F + "," : F;
    }, w.error = function(F, i) {
      return this._error ? this._error(F, i) : new Error(F);
    }, f(c, [{
      key: "errorGenerator",
      set: function(F) {
        this._error = F;
      }
    }]), c;
  }(b.default);
  t.default = o, a.exports = t.default;
})(Tt, Tt.exports);
var wr = Tt.exports, xt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = v(Ut), p = G;
  function v(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, f(n, o);
  }
  function f(n, o) {
    return f = Object.setPrototypeOf || function(c, w) {
      return c.__proto__ = w, c;
    }, f(n, o);
  }
  var g = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var c;
      return c = n.call(this, l) || this, c.type = p.SELECTOR, c;
    }
    return o;
  }(b.default);
  t.default = g, a.exports = t.default;
})(xt, xt.exports);
var br = xt.exports, Pt = { exports: {} };
/*! https://mths.be/cssesc v3.0.0 by @mathias */
var xi = {}, Pi = xi.hasOwnProperty, Ai = function(t, b) {
  if (!t)
    return b;
  var p = {};
  for (var v in b)
    p[v] = Pi.call(t, v) ? t[v] : b[v];
  return p;
}, Bi = /[ -,\.\/:-@\[-\^`\{-~]/, Di = /[ -,\.\/:-@\[\]\^`\{-~]/, Li = /(^|\\+)?(\\[A-F0-9]{1,6})\x20(?![a-fA-F0-9\x20])/g, $t = function a(t, b) {
  b = Ai(b, a.options), b.quotes != "single" && b.quotes != "double" && (b.quotes = "single");
  for (var p = b.quotes == "double" ? '"' : "'", v = b.isIdentifier, y = t.charAt(0), f = "", g = 0, n = t.length; g < n; ) {
    var o = t.charAt(g++), l = o.charCodeAt(), c = void 0;
    if (l < 32 || l > 126) {
      if (l >= 55296 && l <= 56319 && g < n) {
        var w = t.charCodeAt(g++);
        (w & 64512) == 56320 ? l = ((l & 1023) << 10) + (w & 1023) + 65536 : g--;
      }
      c = "\\" + l.toString(16).toUpperCase() + " ";
    } else
      b.escapeEverything ? Bi.test(o) ? c = "\\" + o : c = "\\" + l.toString(16).toUpperCase() + " " : /[\t\n\f\r\x0B]/.test(o) ? c = "\\" + l.toString(16).toUpperCase() + " " : o == "\\" || !v && (o == '"' && p == o || o == "'" && p == o) || v && Di.test(o) ? c = "\\" + o : c = o;
    f += c;
  }
  return v && (/^-[-\d]/.test(f) ? f = "\\-" + f.slice(1) : /\d/.test(y) && (f = "\\3" + y + " " + f.slice(1))), f = f.replace(Li, function(B, F, i) {
    return F && F.length % 2 ? B : (F || "") + i;
  }), !v && b.wrap ? p + f + p : f;
};
$t.options = {
  escapeEverything: !1,
  isIdentifier: !1,
  quotes: "single",
  wrap: !1
};
$t.version = "3.0.0";
var Ht = $t;
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = f(Ht), p = ye, v = f(Le), y = G;
  function f(w) {
    return w && w.__esModule ? w : { default: w };
  }
  function g(w, B) {
    for (var F = 0; F < B.length; F++) {
      var i = B[F];
      i.enumerable = i.enumerable || !1, i.configurable = !0, "value" in i && (i.writable = !0), Object.defineProperty(w, i.key, i);
    }
  }
  function n(w, B, F) {
    return B && g(w.prototype, B), w;
  }
  function o(w, B) {
    w.prototype = Object.create(B.prototype), w.prototype.constructor = w, l(w, B);
  }
  function l(w, B) {
    return l = Object.setPrototypeOf || function(i, d) {
      return i.__proto__ = d, i;
    }, l(w, B);
  }
  var c = /* @__PURE__ */ function(w) {
    o(B, w);
    function B(i) {
      var d;
      return d = w.call(this, i) || this, d.type = y.CLASS, d._constructed = !0, d;
    }
    var F = B.prototype;
    return F.valueToString = function() {
      return "." + w.prototype.valueToString.call(this);
    }, n(B, [{
      key: "value",
      get: function() {
        return this._value;
      },
      set: function(d) {
        if (this._constructed) {
          var S = (0, b.default)(d, {
            isIdentifier: !0
          });
          S !== d ? ((0, p.ensureObject)(this, "raws"), this.raws.value = S) : this.raws && delete this.raws.value;
        }
        this._value = d;
      }
    }]), B;
  }(v.default);
  t.default = c, a.exports = t.default;
})(Pt, Pt.exports);
var _r = Pt.exports, At = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = v(Le), p = G;
  function v(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, f(n, o);
  }
  function f(n, o) {
    return f = Object.setPrototypeOf || function(c, w) {
      return c.__proto__ = w, c;
    }, f(n, o);
  }
  var g = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var c;
      return c = n.call(this, l) || this, c.type = p.COMMENT, c;
    }
    return o;
  }(b.default);
  t.default = g, a.exports = t.default;
})(At, At.exports);
var Tr = At.exports, Bt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = v(Le), p = G;
  function v(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, f(n, o);
  }
  function f(n, o) {
    return f = Object.setPrototypeOf || function(c, w) {
      return c.__proto__ = w, c;
    }, f(n, o);
  }
  var g = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(c) {
      var w;
      return w = n.call(this, c) || this, w.type = p.ID, w;
    }
    var l = o.prototype;
    return l.valueToString = function() {
      return "#" + n.prototype.valueToString.call(this);
    }, o;
  }(b.default);
  t.default = g, a.exports = t.default;
})(Bt, Bt.exports);
var yr = Bt.exports, Dt = { exports: {} }, Lt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = y(Ht), p = ye, v = y(Le);
  function y(c) {
    return c && c.__esModule ? c : { default: c };
  }
  function f(c, w) {
    for (var B = 0; B < w.length; B++) {
      var F = w[B];
      F.enumerable = F.enumerable || !1, F.configurable = !0, "value" in F && (F.writable = !0), Object.defineProperty(c, F.key, F);
    }
  }
  function g(c, w, B) {
    return w && f(c.prototype, w), c;
  }
  function n(c, w) {
    c.prototype = Object.create(w.prototype), c.prototype.constructor = c, o(c, w);
  }
  function o(c, w) {
    return o = Object.setPrototypeOf || function(F, i) {
      return F.__proto__ = i, F;
    }, o(c, w);
  }
  var l = /* @__PURE__ */ function(c) {
    n(w, c);
    function w() {
      return c.apply(this, arguments) || this;
    }
    var B = w.prototype;
    return B.qualifiedName = function(i) {
      return this.namespace ? this.namespaceString + "|" + i : i;
    }, B.valueToString = function() {
      return this.qualifiedName(c.prototype.valueToString.call(this));
    }, g(w, [{
      key: "namespace",
      get: function() {
        return this._namespace;
      },
      set: function(i) {
        if (i === !0 || i === "*" || i === "&") {
          this._namespace = i, this.raws && delete this.raws.namespace;
          return;
        }
        var d = (0, b.default)(i, {
          isIdentifier: !0
        });
        this._namespace = i, d !== i ? ((0, p.ensureObject)(this, "raws"), this.raws.namespace = d) : this.raws && delete this.raws.namespace;
      }
    }, {
      key: "ns",
      get: function() {
        return this._namespace;
      },
      set: function(i) {
        this.namespace = i;
      }
    }, {
      key: "namespaceString",
      get: function() {
        if (this.namespace) {
          var i = this.stringifyProperty("namespace");
          return i === !0 ? "" : i;
        } else
          return "";
      }
    }]), w;
  }(v.default);
  t.default = l, a.exports = t.default;
})(Lt, Lt.exports);
var Gt = Lt.exports;
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = v(Gt), p = G;
  function v(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, f(n, o);
  }
  function f(n, o) {
    return f = Object.setPrototypeOf || function(c, w) {
      return c.__proto__ = w, c;
    }, f(n, o);
  }
  var g = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var c;
      return c = n.call(this, l) || this, c.type = p.TAG, c;
    }
    return o;
  }(b.default);
  t.default = g, a.exports = t.default;
})(Dt, Dt.exports);
var Sr = Dt.exports, Ft = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = v(Le), p = G;
  function v(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, f(n, o);
  }
  function f(n, o) {
    return f = Object.setPrototypeOf || function(c, w) {
      return c.__proto__ = w, c;
    }, f(n, o);
  }
  var g = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var c;
      return c = n.call(this, l) || this, c.type = p.STRING, c;
    }
    return o;
  }(b.default);
  t.default = g, a.exports = t.default;
})(Ft, Ft.exports);
var Er = Ft.exports, qt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = v(Ut), p = G;
  function v(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, f(n, o);
  }
  function f(n, o) {
    return f = Object.setPrototypeOf || function(c, w) {
      return c.__proto__ = w, c;
    }, f(n, o);
  }
  var g = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(c) {
      var w;
      return w = n.call(this, c) || this, w.type = p.PSEUDO, w;
    }
    var l = o.prototype;
    return l.toString = function() {
      var w = this.length ? "(" + this.map(String).join(",") + ")" : "";
      return [this.rawSpaceBefore, this.stringifyProperty("value"), w, this.rawSpaceAfter].join("");
    }, o;
  }(b.default);
  t.default = g, a.exports = t.default;
})(qt, qt.exports);
var Or = qt.exports, jt = {}, Fi = qi;
function qi(a, t) {
  if (mt("noDeprecation"))
    return a;
  var b = !1;
  function p() {
    if (!b) {
      if (mt("throwDeprecation"))
        throw new Error(t);
      mt("traceDeprecation") ? console.trace(t) : console.warn(t), b = !0;
    }
    return a.apply(this, arguments);
  }
  return p;
}
function mt(a) {
  try {
    if (!Ue.localStorage)
      return !1;
  } catch {
    return !1;
  }
  var t = Ue.localStorage[a];
  return t == null ? !1 : String(t).toLowerCase() === "true";
}
(function(a) {
  a.__esModule = !0, a.unescapeValue = d, a.default = void 0;
  var t = f(Ht), b = f(mr), p = f(Gt), v = G, y;
  function f(T) {
    return T && T.__esModule ? T : { default: T };
  }
  function g(T, x) {
    for (var N = 0; N < x.length; N++) {
      var L = x[N];
      L.enumerable = L.enumerable || !1, L.configurable = !0, "value" in L && (L.writable = !0), Object.defineProperty(T, L.key, L);
    }
  }
  function n(T, x, N) {
    return x && g(T.prototype, x), T;
  }
  function o(T, x) {
    T.prototype = Object.create(x.prototype), T.prototype.constructor = T, l(T, x);
  }
  function l(T, x) {
    return l = Object.setPrototypeOf || function(L, A) {
      return L.__proto__ = A, L;
    }, l(T, x);
  }
  var c = Fi, w = /^('|")([^]*)\1$/, B = c(function() {
  }, "Assigning an attribute a value containing characters that might need to be escaped is deprecated. Call attribute.setValue() instead."), F = c(function() {
  }, "Assigning attr.quoted is deprecated and has no effect. Assign to attr.quoteMark instead."), i = c(function() {
  }, "Constructing an Attribute selector with a value without specifying quoteMark is deprecated. Note: The value should be unescaped now.");
  function d(T) {
    var x = !1, N = null, L = T, A = L.match(w);
    return A && (N = A[1], L = A[2]), L = (0, b.default)(L), L !== T && (x = !0), {
      deprecatedUsage: x,
      unescaped: L,
      quoteMark: N
    };
  }
  function S(T) {
    if (T.quoteMark !== void 0 || T.value === void 0)
      return T;
    i();
    var x = d(T.value), N = x.quoteMark, L = x.unescaped;
    return T.raws || (T.raws = {}), T.raws.value === void 0 && (T.raws.value = T.value), T.value = L, T.quoteMark = N, T;
  }
  var O = /* @__PURE__ */ function(T) {
    o(x, T);
    function x(L) {
      var A;
      return L === void 0 && (L = {}), A = T.call(this, S(L)) || this, A.type = v.ATTRIBUTE, A.raws = A.raws || {}, Object.defineProperty(A.raws, "unquoted", {
        get: c(function() {
          return A.value;
        }, "attr.raws.unquoted is deprecated. Call attr.value instead."),
        set: c(function() {
          return A.value;
        }, "Setting attr.raws.unquoted is deprecated and has no effect. attr.value is unescaped by default now.")
      }), A._constructed = !0, A;
    }
    var N = x.prototype;
    return N.getQuotedValue = function(A) {
      A === void 0 && (A = {});
      var D = this._determineQuoteMark(A), Y = E[D], Z = (0, t.default)(this._value, Y);
      return Z;
    }, N._determineQuoteMark = function(A) {
      return A.smart ? this.smartQuoteMark(A) : this.preferredQuoteMark(A);
    }, N.setValue = function(A, D) {
      D === void 0 && (D = {}), this._value = A, this._quoteMark = this._determineQuoteMark(D), this._syncRawValue();
    }, N.smartQuoteMark = function(A) {
      var D = this.value, Y = D.replace(/[^']/g, "").length, Z = D.replace(/[^"]/g, "").length;
      if (Y + Z === 0) {
        var pe = (0, t.default)(D, {
          isIdentifier: !0
        });
        if (pe === D)
          return x.NO_QUOTE;
        var ue = this.preferredQuoteMark(A);
        if (ue === x.NO_QUOTE) {
          var ce = this.quoteMark || A.quoteMark || x.DOUBLE_QUOTE, j = E[ce], he = (0, t.default)(D, j);
          if (he.length < pe.length)
            return ce;
        }
        return ue;
      } else
        return Z === Y ? this.preferredQuoteMark(A) : Z < Y ? x.DOUBLE_QUOTE : x.SINGLE_QUOTE;
    }, N.preferredQuoteMark = function(A) {
      var D = A.preferCurrentQuoteMark ? this.quoteMark : A.quoteMark;
      return D === void 0 && (D = A.preferCurrentQuoteMark ? A.quoteMark : this.quoteMark), D === void 0 && (D = x.DOUBLE_QUOTE), D;
    }, N._syncRawValue = function() {
      var A = (0, t.default)(this._value, E[this.quoteMark]);
      A === this._value ? this.raws && delete this.raws.value : this.raws.value = A;
    }, N._handleEscapes = function(A, D) {
      if (this._constructed) {
        var Y = (0, t.default)(D, {
          isIdentifier: !0
        });
        Y !== D ? this.raws[A] = Y : delete this.raws[A];
      }
    }, N._spacesFor = function(A) {
      var D = {
        before: "",
        after: ""
      }, Y = this.spaces[A] || {}, Z = this.raws.spaces && this.raws.spaces[A] || {};
      return Object.assign(D, Y, Z);
    }, N._stringFor = function(A, D, Y) {
      D === void 0 && (D = A), Y === void 0 && (Y = k);
      var Z = this._spacesFor(D);
      return Y(this.stringifyProperty(A), Z);
    }, N.offsetOf = function(A) {
      var D = 1, Y = this._spacesFor("attribute");
      if (D += Y.before.length, A === "namespace" || A === "ns")
        return this.namespace ? D : -1;
      if (A === "attributeNS" || (D += this.namespaceString.length, this.namespace && (D += 1), A === "attribute"))
        return D;
      D += this.stringifyProperty("attribute").length, D += Y.after.length;
      var Z = this._spacesFor("operator");
      D += Z.before.length;
      var pe = this.stringifyProperty("operator");
      if (A === "operator")
        return pe ? D : -1;
      D += pe.length, D += Z.after.length;
      var ue = this._spacesFor("value");
      D += ue.before.length;
      var ce = this.stringifyProperty("value");
      if (A === "value")
        return ce ? D : -1;
      D += ce.length, D += ue.after.length;
      var j = this._spacesFor("insensitive");
      return D += j.before.length, A === "insensitive" && this.insensitive ? D : -1;
    }, N.toString = function() {
      var A = this, D = [this.rawSpaceBefore, "["];
      return D.push(this._stringFor("qualifiedAttribute", "attribute")), this.operator && (this.value || this.value === "") && (D.push(this._stringFor("operator")), D.push(this._stringFor("value")), D.push(this._stringFor("insensitiveFlag", "insensitive", function(Y, Z) {
        return Y.length > 0 && !A.quoted && Z.before.length === 0 && !(A.spaces.value && A.spaces.value.after) && (Z.before = " "), k(Y, Z);
      }))), D.push("]"), D.push(this.rawSpaceAfter), D.join("");
    }, n(x, [{
      key: "quoted",
      get: function() {
        var A = this.quoteMark;
        return A === "'" || A === '"';
      },
      set: function(A) {
        F();
      }
      /**
       * returns a single (`'`) or double (`"`) quote character if the value is quoted.
       * returns `null` if the value is not quoted.
       * returns `undefined` if the quotation state is unknown (this can happen when
       * the attribute is constructed without specifying a quote mark.)
       */
    }, {
      key: "quoteMark",
      get: function() {
        return this._quoteMark;
      },
      set: function(A) {
        if (!this._constructed) {
          this._quoteMark = A;
          return;
        }
        this._quoteMark !== A && (this._quoteMark = A, this._syncRawValue());
      }
    }, {
      key: "qualifiedAttribute",
      get: function() {
        return this.qualifiedName(this.raws.attribute || this.attribute);
      }
    }, {
      key: "insensitiveFlag",
      get: function() {
        return this.insensitive ? "i" : "";
      }
    }, {
      key: "value",
      get: function() {
        return this._value;
      },
      set: function(A) {
        if (this._constructed) {
          var D = d(A), Y = D.deprecatedUsage, Z = D.unescaped, pe = D.quoteMark;
          if (Y && B(), Z === this._value && pe === this._quoteMark)
            return;
          this._value = Z, this._quoteMark = pe, this._syncRawValue();
        } else
          this._value = A;
      }
    }, {
      key: "attribute",
      get: function() {
        return this._attribute;
      },
      set: function(A) {
        this._handleEscapes("attribute", A), this._attribute = A;
      }
    }]), x;
  }(p.default);
  a.default = O, O.NO_QUOTE = null, O.SINGLE_QUOTE = "'", O.DOUBLE_QUOTE = '"';
  var E = (y = {
    "'": {
      quotes: "single",
      wrap: !0
    },
    '"': {
      quotes: "double",
      wrap: !0
    }
  }, y[null] = {
    isIdentifier: !0
  }, y);
  function k(T, x) {
    return "" + x.before + T + x.after;
  }
})(jt);
var Mt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = v(Gt), p = G;
  function v(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, f(n, o);
  }
  function f(n, o) {
    return f = Object.setPrototypeOf || function(c, w) {
      return c.__proto__ = w, c;
    }, f(n, o);
  }
  var g = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var c;
      return c = n.call(this, l) || this, c.type = p.UNIVERSAL, c.value = "*", c;
    }
    return o;
  }(b.default);
  t.default = g, a.exports = t.default;
})(Mt, Mt.exports);
var Ir = Mt.exports, Rt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = v(Le), p = G;
  function v(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, f(n, o);
  }
  function f(n, o) {
    return f = Object.setPrototypeOf || function(c, w) {
      return c.__proto__ = w, c;
    }, f(n, o);
  }
  var g = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var c;
      return c = n.call(this, l) || this, c.type = p.COMBINATOR, c;
    }
    return o;
  }(b.default);
  t.default = g, a.exports = t.default;
})(Rt, Rt.exports);
var kr = Rt.exports, Ct = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = v(Le), p = G;
  function v(n) {
    return n && n.__esModule ? n : { default: n };
  }
  function y(n, o) {
    n.prototype = Object.create(o.prototype), n.prototype.constructor = n, f(n, o);
  }
  function f(n, o) {
    return f = Object.setPrototypeOf || function(c, w) {
      return c.__proto__ = w, c;
    }, f(n, o);
  }
  var g = /* @__PURE__ */ function(n) {
    y(o, n);
    function o(l) {
      var c;
      return c = n.call(this, l) || this, c.type = p.NESTING, c.value = "&", c;
    }
    return o;
  }(b.default);
  t.default = g, a.exports = t.default;
})(Ct, Ct.exports);
var xr = Ct.exports, Nt = { exports: {} };
(function(a, t) {
  t.__esModule = !0, t.default = b;
  function b(p) {
    return p.sort(function(v, y) {
      return v - y;
    });
  }
  a.exports = t.default;
})(Nt, Nt.exports);
var Mi = Nt.exports, Pr = {}, M = {};
M.__esModule = !0;
M.combinator = M.word = M.comment = M.str = M.tab = M.newline = M.feed = M.cr = M.backslash = M.bang = M.slash = M.doubleQuote = M.singleQuote = M.space = M.greaterThan = M.pipe = M.equals = M.plus = M.caret = M.tilde = M.dollar = M.closeSquare = M.openSquare = M.closeParenthesis = M.openParenthesis = M.semicolon = M.colon = M.comma = M.at = M.asterisk = M.ampersand = void 0;
var Ri = 38;
M.ampersand = Ri;
var Ci = 42;
M.asterisk = Ci;
var Ni = 64;
M.at = Ni;
var zi = 44;
M.comma = zi;
var Wi = 58;
M.colon = Wi;
var Ui = 59;
M.semicolon = Ui;
var $i = 40;
M.openParenthesis = $i;
var Hi = 41;
M.closeParenthesis = Hi;
var Gi = 91;
M.openSquare = Gi;
var ji = 93;
M.closeSquare = ji;
var Qi = 36;
M.dollar = Qi;
var Yi = 126;
M.tilde = Yi;
var Vi = 94;
M.caret = Vi;
var Ki = 43;
M.plus = Ki;
var Ji = 61;
M.equals = Ji;
var Xi = 124;
M.pipe = Xi;
var Zi = 62;
M.greaterThan = Zi;
var eo = 32;
M.space = eo;
var Ar = 39;
M.singleQuote = Ar;
var to = 34;
M.doubleQuote = to;
var ro = 47;
M.slash = ro;
var no = 33;
M.bang = no;
var io = 92;
M.backslash = io;
var oo = 13;
M.cr = oo;
var ao = 12;
M.feed = ao;
var so = 10;
M.newline = so;
var uo = 9;
M.tab = uo;
var lo = Ar;
M.str = lo;
var co = -1;
M.comment = co;
var fo = -2;
M.word = fo;
var po = -3;
M.combinator = po;
(function(a) {
  a.__esModule = !0, a.default = F, a.FIELDS = void 0;
  var t = y(M), b, p;
  function v() {
    if (typeof WeakMap != "function")
      return null;
    var i = /* @__PURE__ */ new WeakMap();
    return v = function() {
      return i;
    }, i;
  }
  function y(i) {
    if (i && i.__esModule)
      return i;
    if (i === null || typeof i != "object" && typeof i != "function")
      return { default: i };
    var d = v();
    if (d && d.has(i))
      return d.get(i);
    var S = {}, O = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var E in i)
      if (Object.prototype.hasOwnProperty.call(i, E)) {
        var k = O ? Object.getOwnPropertyDescriptor(i, E) : null;
        k && (k.get || k.set) ? Object.defineProperty(S, E, k) : S[E] = i[E];
      }
    return S.default = i, d && d.set(i, S), S;
  }
  for (var f = (b = {}, b[t.tab] = !0, b[t.newline] = !0, b[t.cr] = !0, b[t.feed] = !0, b), g = (p = {}, p[t.space] = !0, p[t.tab] = !0, p[t.newline] = !0, p[t.cr] = !0, p[t.feed] = !0, p[t.ampersand] = !0, p[t.asterisk] = !0, p[t.bang] = !0, p[t.comma] = !0, p[t.colon] = !0, p[t.semicolon] = !0, p[t.openParenthesis] = !0, p[t.closeParenthesis] = !0, p[t.openSquare] = !0, p[t.closeSquare] = !0, p[t.singleQuote] = !0, p[t.doubleQuote] = !0, p[t.plus] = !0, p[t.pipe] = !0, p[t.tilde] = !0, p[t.greaterThan] = !0, p[t.equals] = !0, p[t.dollar] = !0, p[t.caret] = !0, p[t.slash] = !0, p), n = {}, o = "0123456789abcdefABCDEF", l = 0; l < o.length; l++)
    n[o.charCodeAt(l)] = !0;
  function c(i, d) {
    var S = d, O;
    do {
      if (O = i.charCodeAt(S), g[O])
        return S - 1;
      O === t.backslash ? S = w(i, S) + 1 : S++;
    } while (S < i.length);
    return S - 1;
  }
  function w(i, d) {
    var S = d, O = i.charCodeAt(S + 1);
    if (!f[O])
      if (n[O]) {
        var E = 0;
        do
          S++, E++, O = i.charCodeAt(S + 1);
        while (n[O] && E < 6);
        E < 6 && O === t.space && S++;
      } else
        S++;
    return S;
  }
  var B = {
    TYPE: 0,
    START_LINE: 1,
    START_COL: 2,
    END_LINE: 3,
    END_COL: 4,
    START_POS: 5,
    END_POS: 6
  };
  a.FIELDS = B;
  function F(i) {
    var d = [], S = i.css.valueOf(), O = S, E = O.length, k = -1, T = 1, x = 0, N = 0, L, A, D, Y, Z, pe, ue, ce, j, he, Se, Fe, Ee;
    function q(R, I) {
      if (i.safe)
        S += I, j = S.length - 1;
      else
        throw i.error("Unclosed " + R, T, x - k, x);
    }
    for (; x < E; ) {
      switch (L = S.charCodeAt(x), L === t.newline && (k = x, T += 1), L) {
        case t.space:
        case t.tab:
        case t.newline:
        case t.cr:
        case t.feed:
          j = x;
          do
            j += 1, L = S.charCodeAt(j), L === t.newline && (k = j, T += 1);
          while (L === t.space || L === t.newline || L === t.tab || L === t.cr || L === t.feed);
          Ee = t.space, Y = T, D = j - k - 1, N = j;
          break;
        case t.plus:
        case t.greaterThan:
        case t.tilde:
        case t.pipe:
          j = x;
          do
            j += 1, L = S.charCodeAt(j);
          while (L === t.plus || L === t.greaterThan || L === t.tilde || L === t.pipe);
          Ee = t.combinator, Y = T, D = x - k, N = j;
          break;
        case t.asterisk:
        case t.ampersand:
        case t.bang:
        case t.comma:
        case t.equals:
        case t.dollar:
        case t.caret:
        case t.openSquare:
        case t.closeSquare:
        case t.colon:
        case t.semicolon:
        case t.openParenthesis:
        case t.closeParenthesis:
          j = x, Ee = L, Y = T, D = x - k, N = j + 1;
          break;
        case t.singleQuote:
        case t.doubleQuote:
          Fe = L === t.singleQuote ? "'" : '"', j = x;
          do
            for (Z = !1, j = S.indexOf(Fe, j + 1), j === -1 && q("quote", Fe), pe = j; S.charCodeAt(pe - 1) === t.backslash; )
              pe -= 1, Z = !Z;
          while (Z);
          Ee = t.str, Y = T, D = x - k, N = j + 1;
          break;
        default:
          L === t.slash && S.charCodeAt(x + 1) === t.asterisk ? (j = S.indexOf("*/", x + 2) + 1, j === 0 && q("comment", "*/"), A = S.slice(x, j + 1), ce = A.split(`
`), ue = ce.length - 1, ue > 0 ? (he = T + ue, Se = j - ce[ue].length) : (he = T, Se = k), Ee = t.comment, T = he, Y = he, D = j - Se) : L === t.slash ? (j = x, Ee = L, Y = T, D = x - k, N = j + 1) : (j = c(S, x), Ee = t.word, Y = T, D = j - k), N = j + 1;
          break;
      }
      d.push([
        Ee,
        // [0] Token type
        T,
        // [1] Starting line
        x - k,
        // [2] Starting column
        Y,
        // [3] Ending line
        D,
        // [4] Ending column
        x,
        // [5] Start position / Source index
        N
        // [6] End position
      ]), Se && (k = Se, Se = null), x = N;
    }
    return d;
  }
})(Pr);
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = N(wr), p = N(br), v = N(_r), y = N(Tr), f = N(yr), g = N(Sr), n = N(Er), o = N(Or), l = x(jt), c = N(Ir), w = N(kr), B = N(xr), F = N(Mi), i = x(Pr), d = x(M), S = x(G), O = ye, E, k;
  function T() {
    if (typeof WeakMap != "function")
      return null;
    var q = /* @__PURE__ */ new WeakMap();
    return T = function() {
      return q;
    }, q;
  }
  function x(q) {
    if (q && q.__esModule)
      return q;
    if (q === null || typeof q != "object" && typeof q != "function")
      return { default: q };
    var R = T();
    if (R && R.has(q))
      return R.get(q);
    var I = {}, m = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var P in q)
      if (Object.prototype.hasOwnProperty.call(q, P)) {
        var z = m ? Object.getOwnPropertyDescriptor(q, P) : null;
        z && (z.get || z.set) ? Object.defineProperty(I, P, z) : I[P] = q[P];
      }
    return I.default = q, R && R.set(q, I), I;
  }
  function N(q) {
    return q && q.__esModule ? q : { default: q };
  }
  function L(q, R) {
    for (var I = 0; I < R.length; I++) {
      var m = R[I];
      m.enumerable = m.enumerable || !1, m.configurable = !0, "value" in m && (m.writable = !0), Object.defineProperty(q, m.key, m);
    }
  }
  function A(q, R, I) {
    return R && L(q.prototype, R), q;
  }
  var D = (E = {}, E[d.space] = !0, E[d.cr] = !0, E[d.feed] = !0, E[d.newline] = !0, E[d.tab] = !0, E), Y = Object.assign({}, D, (k = {}, k[d.comment] = !0, k));
  function Z(q) {
    return {
      line: q[i.FIELDS.START_LINE],
      column: q[i.FIELDS.START_COL]
    };
  }
  function pe(q) {
    return {
      line: q[i.FIELDS.END_LINE],
      column: q[i.FIELDS.END_COL]
    };
  }
  function ue(q, R, I, m) {
    return {
      start: {
        line: q,
        column: R
      },
      end: {
        line: I,
        column: m
      }
    };
  }
  function ce(q) {
    return ue(q[i.FIELDS.START_LINE], q[i.FIELDS.START_COL], q[i.FIELDS.END_LINE], q[i.FIELDS.END_COL]);
  }
  function j(q, R) {
    if (q)
      return ue(q[i.FIELDS.START_LINE], q[i.FIELDS.START_COL], R[i.FIELDS.END_LINE], R[i.FIELDS.END_COL]);
  }
  function he(q, R) {
    var I = q[R];
    if (typeof I == "string")
      return I.indexOf("\\") !== -1 && ((0, O.ensureObject)(q, "raws"), q[R] = (0, O.unesc)(I), q.raws[R] === void 0 && (q.raws[R] = I)), q;
  }
  function Se(q, R) {
    for (var I = -1, m = []; (I = q.indexOf(R, I + 1)) !== -1; )
      m.push(I);
    return m;
  }
  function Fe() {
    var q = Array.prototype.concat.apply([], arguments);
    return q.filter(function(R, I) {
      return I === q.indexOf(R);
    });
  }
  var Ee = /* @__PURE__ */ function() {
    function q(I, m) {
      m === void 0 && (m = {}), this.rule = I, this.options = Object.assign({
        lossy: !1,
        safe: !1
      }, m), this.position = 0, this.css = typeof this.rule == "string" ? this.rule : this.rule.selector, this.tokens = (0, i.default)({
        css: this.css,
        error: this._errorGenerator(),
        safe: this.options.safe
      });
      var P = j(this.tokens[0], this.tokens[this.tokens.length - 1]);
      this.root = new b.default({
        source: P
      }), this.root.errorGenerator = this._errorGenerator();
      var z = new p.default({
        source: {
          start: {
            line: 1,
            column: 1
          }
        }
      });
      this.root.append(z), this.current = z, this.loop();
    }
    var R = q.prototype;
    return R._errorGenerator = function() {
      var m = this;
      return function(P, z) {
        return typeof m.rule == "string" ? new Error(P) : m.rule.error(P, z);
      };
    }, R.attribute = function() {
      var m = [], P = this.currToken;
      for (this.position++; this.position < this.tokens.length && this.currToken[i.FIELDS.TYPE] !== d.closeSquare; )
        m.push(this.currToken), this.position++;
      if (this.currToken[i.FIELDS.TYPE] !== d.closeSquare)
        return this.expected("closing square bracket", this.currToken[i.FIELDS.START_POS]);
      var z = m.length, _ = {
        source: ue(P[1], P[2], this.currToken[3], this.currToken[4]),
        sourceIndex: P[i.FIELDS.START_POS]
      };
      if (z === 1 && !~[d.word].indexOf(m[0][i.FIELDS.TYPE]))
        return this.expected("attribute", m[0][i.FIELDS.START_POS]);
      for (var $ = 0, K = "", Q = "", W = null, ee = !1; $ < z; ) {
        var le = m[$], H = this.content(le), ie = m[$ + 1];
        switch (le[i.FIELDS.TYPE]) {
          case d.space:
            if (ee = !0, this.options.lossy)
              break;
            if (W) {
              (0, O.ensureObject)(_, "spaces", W);
              var xe = _.spaces[W].after || "";
              _.spaces[W].after = xe + H;
              var Oe = (0, O.getProp)(_, "raws", "spaces", W, "after") || null;
              Oe && (_.raws.spaces[W].after = Oe + H);
            } else
              K = K + H, Q = Q + H;
            break;
          case d.asterisk:
            if (ie[i.FIELDS.TYPE] === d.equals)
              _.operator = H, W = "operator";
            else if ((!_.namespace || W === "namespace" && !ee) && ie) {
              K && ((0, O.ensureObject)(_, "spaces", "attribute"), _.spaces.attribute.before = K, K = ""), Q && ((0, O.ensureObject)(_, "raws", "spaces", "attribute"), _.raws.spaces.attribute.before = K, Q = ""), _.namespace = (_.namespace || "") + H;
              var ge = (0, O.getProp)(_, "raws", "namespace") || null;
              ge && (_.raws.namespace += H), W = "namespace";
            }
            ee = !1;
            break;
          case d.dollar:
            if (W === "value") {
              var fe = (0, O.getProp)(_, "raws", "value");
              _.value += "$", fe && (_.raws.value = fe + "$");
              break;
            }
          case d.caret:
            ie[i.FIELDS.TYPE] === d.equals && (_.operator = H, W = "operator"), ee = !1;
            break;
          case d.combinator:
            if (H === "~" && ie[i.FIELDS.TYPE] === d.equals && (_.operator = H, W = "operator"), H !== "|") {
              ee = !1;
              break;
            }
            ie[i.FIELDS.TYPE] === d.equals ? (_.operator = H, W = "operator") : !_.namespace && !_.attribute && (_.namespace = !0), ee = !1;
            break;
          case d.word:
            if (ie && this.content(ie) === "|" && m[$ + 2] && m[$ + 2][i.FIELDS.TYPE] !== d.equals && // this look-ahead probably fails with comment nodes involved.
            !_.operator && !_.namespace)
              _.namespace = H, W = "namespace";
            else if (!_.attribute || W === "attribute" && !ee) {
              K && ((0, O.ensureObject)(_, "spaces", "attribute"), _.spaces.attribute.before = K, K = ""), Q && ((0, O.ensureObject)(_, "raws", "spaces", "attribute"), _.raws.spaces.attribute.before = Q, Q = ""), _.attribute = (_.attribute || "") + H;
              var de = (0, O.getProp)(_, "raws", "attribute") || null;
              de && (_.raws.attribute += H), W = "attribute";
            } else if (!_.value && _.value !== "" || W === "value" && !ee) {
              var ve = (0, O.unesc)(H), be = (0, O.getProp)(_, "raws", "value") || "", _e = _.value || "";
              _.value = _e + ve, _.quoteMark = null, (ve !== H || be) && ((0, O.ensureObject)(_, "raws"), _.raws.value = (be || _e) + H), W = "value";
            } else {
              var qe = H === "i" || H === "I";
              (_.value || _.value === "") && (_.quoteMark || ee) ? (_.insensitive = qe, (!qe || H === "I") && ((0, O.ensureObject)(_, "raws"), _.raws.insensitiveFlag = H), W = "insensitive", K && ((0, O.ensureObject)(_, "spaces", "insensitive"), _.spaces.insensitive.before = K, K = ""), Q && ((0, O.ensureObject)(_, "raws", "spaces", "insensitive"), _.raws.spaces.insensitive.before = Q, Q = "")) : (_.value || _.value === "") && (W = "value", _.value += H, _.raws.value && (_.raws.value += H));
            }
            ee = !1;
            break;
          case d.str:
            if (!_.attribute || !_.operator)
              return this.error("Expected an attribute followed by an operator preceding the string.", {
                index: le[i.FIELDS.START_POS]
              });
            var De = (0, l.unescapeValue)(H), rt = De.unescaped, nt = De.quoteMark;
            _.value = rt, _.quoteMark = nt, W = "value", (0, O.ensureObject)(_, "raws"), _.raws.value = H, ee = !1;
            break;
          case d.equals:
            if (!_.attribute)
              return this.expected("attribute", le[i.FIELDS.START_POS], H);
            if (_.value)
              return this.error('Unexpected "=" found; an operator was already defined.', {
                index: le[i.FIELDS.START_POS]
              });
            _.operator = _.operator ? _.operator + H : H, W = "operator", ee = !1;
            break;
          case d.comment:
            if (W)
              if (ee || ie && ie[i.FIELDS.TYPE] === d.space || W === "insensitive") {
                var Ne = (0, O.getProp)(_, "spaces", W, "after") || "", je = (0, O.getProp)(_, "raws", "spaces", W, "after") || Ne;
                (0, O.ensureObject)(_, "raws", "spaces", W), _.raws.spaces[W].after = je + H;
              } else {
                var Qe = _[W] || "", Ye = (0, O.getProp)(_, "raws", W) || Qe;
                (0, O.ensureObject)(_, "raws"), _.raws[W] = Ye + H;
              }
            else
              Q = Q + H;
            break;
          default:
            return this.error('Unexpected "' + H + '" found.', {
              index: le[i.FIELDS.START_POS]
            });
        }
        $++;
      }
      he(_, "attribute"), he(_, "namespace"), this.newNode(new l.default(_)), this.position++;
    }, R.parseWhitespaceEquivalentTokens = function(m) {
      m < 0 && (m = this.tokens.length);
      var P = this.position, z = [], _ = "", $ = void 0;
      do
        if (D[this.currToken[i.FIELDS.TYPE]])
          this.options.lossy || (_ += this.content());
        else if (this.currToken[i.FIELDS.TYPE] === d.comment) {
          var K = {};
          _ && (K.before = _, _ = ""), $ = new y.default({
            value: this.content(),
            source: ce(this.currToken),
            sourceIndex: this.currToken[i.FIELDS.START_POS],
            spaces: K
          }), z.push($);
        }
      while (++this.position < m);
      if (_) {
        if ($)
          $.spaces.after = _;
        else if (!this.options.lossy) {
          var Q = this.tokens[P], W = this.tokens[this.position - 1];
          z.push(new n.default({
            value: "",
            source: ue(Q[i.FIELDS.START_LINE], Q[i.FIELDS.START_COL], W[i.FIELDS.END_LINE], W[i.FIELDS.END_COL]),
            sourceIndex: Q[i.FIELDS.START_POS],
            spaces: {
              before: _,
              after: ""
            }
          }));
        }
      }
      return z;
    }, R.convertWhitespaceNodesToSpace = function(m, P) {
      var z = this;
      P === void 0 && (P = !1);
      var _ = "", $ = "";
      m.forEach(function(Q) {
        var W = z.lossySpace(Q.spaces.before, P), ee = z.lossySpace(Q.rawSpaceBefore, P);
        _ += W + z.lossySpace(Q.spaces.after, P && W.length === 0), $ += W + Q.value + z.lossySpace(Q.rawSpaceAfter, P && ee.length === 0);
      }), $ === _ && ($ = void 0);
      var K = {
        space: _,
        rawSpace: $
      };
      return K;
    }, R.isNamedCombinator = function(m) {
      return m === void 0 && (m = this.position), this.tokens[m + 0] && this.tokens[m + 0][i.FIELDS.TYPE] === d.slash && this.tokens[m + 1] && this.tokens[m + 1][i.FIELDS.TYPE] === d.word && this.tokens[m + 2] && this.tokens[m + 2][i.FIELDS.TYPE] === d.slash;
    }, R.namedCombinator = function() {
      if (this.isNamedCombinator()) {
        var m = this.content(this.tokens[this.position + 1]), P = (0, O.unesc)(m).toLowerCase(), z = {};
        P !== m && (z.value = "/" + m + "/");
        var _ = new w.default({
          value: "/" + P + "/",
          source: ue(this.currToken[i.FIELDS.START_LINE], this.currToken[i.FIELDS.START_COL], this.tokens[this.position + 2][i.FIELDS.END_LINE], this.tokens[this.position + 2][i.FIELDS.END_COL]),
          sourceIndex: this.currToken[i.FIELDS.START_POS],
          raws: z
        });
        return this.position = this.position + 3, _;
      } else
        this.unexpected();
    }, R.combinator = function() {
      var m = this;
      if (this.content() === "|")
        return this.namespace();
      var P = this.locateNextMeaningfulToken(this.position);
      if (P < 0 || this.tokens[P][i.FIELDS.TYPE] === d.comma) {
        var z = this.parseWhitespaceEquivalentTokens(P);
        if (z.length > 0) {
          var _ = this.current.last;
          if (_) {
            var $ = this.convertWhitespaceNodesToSpace(z), K = $.space, Q = $.rawSpace;
            Q !== void 0 && (_.rawSpaceAfter += Q), _.spaces.after += K;
          } else
            z.forEach(function(be) {
              return m.newNode(be);
            });
        }
        return;
      }
      var W = this.currToken, ee = void 0;
      P > this.position && (ee = this.parseWhitespaceEquivalentTokens(P));
      var le;
      if (this.isNamedCombinator() ? le = this.namedCombinator() : this.currToken[i.FIELDS.TYPE] === d.combinator ? (le = new w.default({
        value: this.content(),
        source: ce(this.currToken),
        sourceIndex: this.currToken[i.FIELDS.START_POS]
      }), this.position++) : D[this.currToken[i.FIELDS.TYPE]] || ee || this.unexpected(), le) {
        if (ee) {
          var H = this.convertWhitespaceNodesToSpace(ee), ie = H.space, xe = H.rawSpace;
          le.spaces.before = ie, le.rawSpaceBefore = xe;
        }
      } else {
        var Oe = this.convertWhitespaceNodesToSpace(ee, !0), ge = Oe.space, fe = Oe.rawSpace;
        fe || (fe = ge);
        var de = {}, ve = {
          spaces: {}
        };
        ge.endsWith(" ") && fe.endsWith(" ") ? (de.before = ge.slice(0, ge.length - 1), ve.spaces.before = fe.slice(0, fe.length - 1)) : ge.startsWith(" ") && fe.startsWith(" ") ? (de.after = ge.slice(1), ve.spaces.after = fe.slice(1)) : ve.value = fe, le = new w.default({
          value: " ",
          source: j(W, this.tokens[this.position - 1]),
          sourceIndex: W[i.FIELDS.START_POS],
          spaces: de,
          raws: ve
        });
      }
      return this.currToken && this.currToken[i.FIELDS.TYPE] === d.space && (le.spaces.after = this.optionalSpace(this.content()), this.position++), this.newNode(le);
    }, R.comma = function() {
      if (this.position === this.tokens.length - 1) {
        this.root.trailingComma = !0, this.position++;
        return;
      }
      this.current._inferEndPosition();
      var m = new p.default({
        source: {
          start: Z(this.tokens[this.position + 1])
        }
      });
      this.current.parent.append(m), this.current = m, this.position++;
    }, R.comment = function() {
      var m = this.currToken;
      this.newNode(new y.default({
        value: this.content(),
        source: ce(m),
        sourceIndex: m[i.FIELDS.START_POS]
      })), this.position++;
    }, R.error = function(m, P) {
      throw this.root.error(m, P);
    }, R.missingBackslash = function() {
      return this.error("Expected a backslash preceding the semicolon.", {
        index: this.currToken[i.FIELDS.START_POS]
      });
    }, R.missingParenthesis = function() {
      return this.expected("opening parenthesis", this.currToken[i.FIELDS.START_POS]);
    }, R.missingSquareBracket = function() {
      return this.expected("opening square bracket", this.currToken[i.FIELDS.START_POS]);
    }, R.unexpected = function() {
      return this.error("Unexpected '" + this.content() + "'. Escaping special characters with \\ may help.", this.currToken[i.FIELDS.START_POS]);
    }, R.namespace = function() {
      var m = this.prevToken && this.content(this.prevToken) || !0;
      if (this.nextToken[i.FIELDS.TYPE] === d.word)
        return this.position++, this.word(m);
      if (this.nextToken[i.FIELDS.TYPE] === d.asterisk)
        return this.position++, this.universal(m);
    }, R.nesting = function() {
      if (this.nextToken) {
        var m = this.content(this.nextToken);
        if (m === "|") {
          this.position++;
          return;
        }
      }
      var P = this.currToken;
      this.newNode(new B.default({
        value: this.content(),
        source: ce(P),
        sourceIndex: P[i.FIELDS.START_POS]
      })), this.position++;
    }, R.parentheses = function() {
      var m = this.current.last, P = 1;
      if (this.position++, m && m.type === S.PSEUDO) {
        var z = new p.default({
          source: {
            start: Z(this.tokens[this.position - 1])
          }
        }), _ = this.current;
        for (m.append(z), this.current = z; this.position < this.tokens.length && P; )
          this.currToken[i.FIELDS.TYPE] === d.openParenthesis && P++, this.currToken[i.FIELDS.TYPE] === d.closeParenthesis && P--, P ? this.parse() : (this.current.source.end = pe(this.currToken), this.current.parent.source.end = pe(this.currToken), this.position++);
        this.current = _;
      } else {
        for (var $ = this.currToken, K = "(", Q; this.position < this.tokens.length && P; )
          this.currToken[i.FIELDS.TYPE] === d.openParenthesis && P++, this.currToken[i.FIELDS.TYPE] === d.closeParenthesis && P--, Q = this.currToken, K += this.parseParenthesisToken(this.currToken), this.position++;
        m ? m.appendToPropertyAndEscape("value", K, K) : this.newNode(new n.default({
          value: K,
          source: ue($[i.FIELDS.START_LINE], $[i.FIELDS.START_COL], Q[i.FIELDS.END_LINE], Q[i.FIELDS.END_COL]),
          sourceIndex: $[i.FIELDS.START_POS]
        }));
      }
      if (P)
        return this.expected("closing parenthesis", this.currToken[i.FIELDS.START_POS]);
    }, R.pseudo = function() {
      for (var m = this, P = "", z = this.currToken; this.currToken && this.currToken[i.FIELDS.TYPE] === d.colon; )
        P += this.content(), this.position++;
      if (!this.currToken)
        return this.expected(["pseudo-class", "pseudo-element"], this.position - 1);
      if (this.currToken[i.FIELDS.TYPE] === d.word)
        this.splitWord(!1, function(_, $) {
          P += _, m.newNode(new o.default({
            value: P,
            source: j(z, m.currToken),
            sourceIndex: z[i.FIELDS.START_POS]
          })), $ > 1 && m.nextToken && m.nextToken[i.FIELDS.TYPE] === d.openParenthesis && m.error("Misplaced parenthesis.", {
            index: m.nextToken[i.FIELDS.START_POS]
          });
        });
      else
        return this.expected(["pseudo-class", "pseudo-element"], this.currToken[i.FIELDS.START_POS]);
    }, R.space = function() {
      var m = this.content();
      this.position === 0 || this.prevToken[i.FIELDS.TYPE] === d.comma || this.prevToken[i.FIELDS.TYPE] === d.openParenthesis || this.current.nodes.every(function(P) {
        return P.type === "comment";
      }) ? (this.spaces = this.optionalSpace(m), this.position++) : this.position === this.tokens.length - 1 || this.nextToken[i.FIELDS.TYPE] === d.comma || this.nextToken[i.FIELDS.TYPE] === d.closeParenthesis ? (this.current.last.spaces.after = this.optionalSpace(m), this.position++) : this.combinator();
    }, R.string = function() {
      var m = this.currToken;
      this.newNode(new n.default({
        value: this.content(),
        source: ce(m),
        sourceIndex: m[i.FIELDS.START_POS]
      })), this.position++;
    }, R.universal = function(m) {
      var P = this.nextToken;
      if (P && this.content(P) === "|")
        return this.position++, this.namespace();
      var z = this.currToken;
      this.newNode(new c.default({
        value: this.content(),
        source: ce(z),
        sourceIndex: z[i.FIELDS.START_POS]
      }), m), this.position++;
    }, R.splitWord = function(m, P) {
      for (var z = this, _ = this.nextToken, $ = this.content(); _ && ~[d.dollar, d.caret, d.equals, d.word].indexOf(_[i.FIELDS.TYPE]); ) {
        this.position++;
        var K = this.content();
        if ($ += K, K.lastIndexOf("\\") === K.length - 1) {
          var Q = this.nextToken;
          Q && Q[i.FIELDS.TYPE] === d.space && ($ += this.requiredSpace(this.content(Q)), this.position++);
        }
        _ = this.nextToken;
      }
      var W = Se($, ".").filter(function(ie) {
        var xe = $[ie - 1] === "\\", Oe = /^\d+\.\d+%$/.test($);
        return !xe && !Oe;
      }), ee = Se($, "#").filter(function(ie) {
        return $[ie - 1] !== "\\";
      }), le = Se($, "#{");
      le.length && (ee = ee.filter(function(ie) {
        return !~le.indexOf(ie);
      }));
      var H = (0, F.default)(Fe([0].concat(W, ee)));
      H.forEach(function(ie, xe) {
        var Oe = H[xe + 1] || $.length, ge = $.slice(ie, Oe);
        if (xe === 0 && P)
          return P.call(z, ge, H.length);
        var fe, de = z.currToken, ve = de[i.FIELDS.START_POS] + H[xe], be = ue(de[1], de[2] + ie, de[3], de[2] + (Oe - 1));
        if (~W.indexOf(ie)) {
          var _e = {
            value: ge.slice(1),
            source: be,
            sourceIndex: ve
          };
          fe = new v.default(he(_e, "value"));
        } else if (~ee.indexOf(ie)) {
          var qe = {
            value: ge.slice(1),
            source: be,
            sourceIndex: ve
          };
          fe = new f.default(he(qe, "value"));
        } else {
          var De = {
            value: ge,
            source: be,
            sourceIndex: ve
          };
          he(De, "value"), fe = new g.default(De);
        }
        z.newNode(fe, m), m = null;
      }), this.position++;
    }, R.word = function(m) {
      var P = this.nextToken;
      return P && this.content(P) === "|" ? (this.position++, this.namespace()) : this.splitWord(m);
    }, R.loop = function() {
      for (; this.position < this.tokens.length; )
        this.parse(!0);
      return this.current._inferEndPosition(), this.root;
    }, R.parse = function(m) {
      switch (this.currToken[i.FIELDS.TYPE]) {
        case d.space:
          this.space();
          break;
        case d.comment:
          this.comment();
          break;
        case d.openParenthesis:
          this.parentheses();
          break;
        case d.closeParenthesis:
          m && this.missingParenthesis();
          break;
        case d.openSquare:
          this.attribute();
          break;
        case d.dollar:
        case d.caret:
        case d.equals:
        case d.word:
          this.word();
          break;
        case d.colon:
          this.pseudo();
          break;
        case d.comma:
          this.comma();
          break;
        case d.asterisk:
          this.universal();
          break;
        case d.ampersand:
          this.nesting();
          break;
        case d.slash:
        case d.combinator:
          this.combinator();
          break;
        case d.str:
          this.string();
          break;
        case d.closeSquare:
          this.missingSquareBracket();
        case d.semicolon:
          this.missingBackslash();
        default:
          this.unexpected();
      }
    }, R.expected = function(m, P, z) {
      if (Array.isArray(m)) {
        var _ = m.pop();
        m = m.join(", ") + " or " + _;
      }
      var $ = /^[aeiou]/.test(m[0]) ? "an" : "a";
      return z ? this.error("Expected " + $ + " " + m + ', found "' + z + '" instead.', {
        index: P
      }) : this.error("Expected " + $ + " " + m + ".", {
        index: P
      });
    }, R.requiredSpace = function(m) {
      return this.options.lossy ? " " : m;
    }, R.optionalSpace = function(m) {
      return this.options.lossy ? "" : m;
    }, R.lossySpace = function(m, P) {
      return this.options.lossy ? P ? " " : "" : m;
    }, R.parseParenthesisToken = function(m) {
      var P = this.content(m);
      return m[i.FIELDS.TYPE] === d.space ? this.requiredSpace(P) : P;
    }, R.newNode = function(m, P) {
      return P && (/^ +$/.test(P) && (this.options.lossy || (this.spaces = (this.spaces || "") + P), P = !0), m.namespace = P, he(m, "namespace")), this.spaces && (m.spaces.before = this.spaces, this.spaces = ""), this.current.append(m);
    }, R.content = function(m) {
      return m === void 0 && (m = this.currToken), this.css.slice(m[i.FIELDS.START_POS], m[i.FIELDS.END_POS]);
    }, R.locateNextMeaningfulToken = function(m) {
      m === void 0 && (m = this.position + 1);
      for (var P = m; P < this.tokens.length; )
        if (Y[this.tokens[P][i.FIELDS.TYPE]]) {
          P++;
          continue;
        } else
          return P;
      return -1;
    }, A(q, [{
      key: "currToken",
      get: function() {
        return this.tokens[this.position];
      }
    }, {
      key: "nextToken",
      get: function() {
        return this.tokens[this.position + 1];
      }
    }, {
      key: "prevToken",
      get: function() {
        return this.tokens[this.position - 1];
      }
    }]), q;
  }();
  t.default = Ee, a.exports = t.default;
})(_t, _t.exports);
var ho = _t.exports;
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = p(ho);
  function p(y) {
    return y && y.__esModule ? y : { default: y };
  }
  var v = /* @__PURE__ */ function() {
    function y(g, n) {
      this.func = g || function() {
      }, this.funcRes = null, this.options = n;
    }
    var f = y.prototype;
    return f._shouldUpdateSelector = function(n, o) {
      o === void 0 && (o = {});
      var l = Object.assign({}, this.options, o);
      return l.updateSelector === !1 ? !1 : typeof n != "string";
    }, f._isLossy = function(n) {
      n === void 0 && (n = {});
      var o = Object.assign({}, this.options, n);
      return o.lossless === !1;
    }, f._root = function(n, o) {
      o === void 0 && (o = {});
      var l = new b.default(n, this._parseOptions(o));
      return l.root;
    }, f._parseOptions = function(n) {
      return {
        lossy: this._isLossy(n)
      };
    }, f._run = function(n, o) {
      var l = this;
      return o === void 0 && (o = {}), new Promise(function(c, w) {
        try {
          var B = l._root(n, o);
          Promise.resolve(l.func(B)).then(function(F) {
            var i = void 0;
            return l._shouldUpdateSelector(n, o) && (i = B.toString(), n.selector = i), {
              transform: F,
              root: B,
              string: i
            };
          }).then(c, w);
        } catch (F) {
          w(F);
          return;
        }
      });
    }, f._runSync = function(n, o) {
      o === void 0 && (o = {});
      var l = this._root(n, o), c = this.func(l);
      if (c && typeof c.then == "function")
        throw new Error("Selector processor returned a promise to a synchronous call.");
      var w = void 0;
      return o.updateSelector && typeof n != "string" && (w = l.toString(), n.selector = w), {
        transform: c,
        root: l,
        string: w
      };
    }, f.ast = function(n, o) {
      return this._run(n, o).then(function(l) {
        return l.root;
      });
    }, f.astSync = function(n, o) {
      return this._runSync(n, o).root;
    }, f.transform = function(n, o) {
      return this._run(n, o).then(function(l) {
        return l.transform;
      });
    }, f.transformSync = function(n, o) {
      return this._runSync(n, o).transform;
    }, f.process = function(n, o) {
      return this._run(n, o).then(function(l) {
        return l.string || l.root.toString();
      });
    }, f.processSync = function(n, o) {
      var l = this._runSync(n, o);
      return l.string || l.root.toString();
    }, y;
  }();
  t.default = v, a.exports = t.default;
})(bt, bt.exports);
var go = bt.exports, Br = {}, re = {};
re.__esModule = !0;
re.universal = re.tag = re.string = re.selector = re.root = re.pseudo = re.nesting = re.id = re.comment = re.combinator = re.className = re.attribute = void 0;
var vo = Ie(jt), mo = Ie(_r), wo = Ie(kr), bo = Ie(Tr), _o = Ie(yr), To = Ie(xr), yo = Ie(Or), So = Ie(wr), Eo = Ie(br), Oo = Ie(Er), Io = Ie(Sr), ko = Ie(Ir);
function Ie(a) {
  return a && a.__esModule ? a : { default: a };
}
var xo = function(t) {
  return new vo.default(t);
};
re.attribute = xo;
var Po = function(t) {
  return new mo.default(t);
};
re.className = Po;
var Ao = function(t) {
  return new wo.default(t);
};
re.combinator = Ao;
var Bo = function(t) {
  return new bo.default(t);
};
re.comment = Bo;
var Do = function(t) {
  return new _o.default(t);
};
re.id = Do;
var Lo = function(t) {
  return new To.default(t);
};
re.nesting = Lo;
var Fo = function(t) {
  return new yo.default(t);
};
re.pseudo = Fo;
var qo = function(t) {
  return new So.default(t);
};
re.root = qo;
var Mo = function(t) {
  return new Eo.default(t);
};
re.selector = Mo;
var Ro = function(t) {
  return new Oo.default(t);
};
re.string = Ro;
var Co = function(t) {
  return new Io.default(t);
};
re.tag = Co;
var No = function(t) {
  return new ko.default(t);
};
re.universal = No;
var X = {};
X.__esModule = !0;
X.isNode = Qt;
X.isPseudoElement = Fr;
X.isPseudoClass = Ko;
X.isContainer = Jo;
X.isNamespace = Xo;
X.isUniversal = X.isTag = X.isString = X.isSelector = X.isRoot = X.isPseudo = X.isNesting = X.isIdentifier = X.isComment = X.isCombinator = X.isClassName = X.isAttribute = void 0;
var se = G, we, zo = (we = {}, we[se.ATTRIBUTE] = !0, we[se.CLASS] = !0, we[se.COMBINATOR] = !0, we[se.COMMENT] = !0, we[se.ID] = !0, we[se.NESTING] = !0, we[se.PSEUDO] = !0, we[se.ROOT] = !0, we[se.SELECTOR] = !0, we[se.STRING] = !0, we[se.TAG] = !0, we[se.UNIVERSAL] = !0, we);
function Qt(a) {
  return typeof a == "object" && zo[a.type];
}
function ke(a, t) {
  return Qt(t) && t.type === a;
}
var Dr = ke.bind(null, se.ATTRIBUTE);
X.isAttribute = Dr;
var Wo = ke.bind(null, se.CLASS);
X.isClassName = Wo;
var Uo = ke.bind(null, se.COMBINATOR);
X.isCombinator = Uo;
var $o = ke.bind(null, se.COMMENT);
X.isComment = $o;
var Ho = ke.bind(null, se.ID);
X.isIdentifier = Ho;
var Go = ke.bind(null, se.NESTING);
X.isNesting = Go;
var Yt = ke.bind(null, se.PSEUDO);
X.isPseudo = Yt;
var jo = ke.bind(null, se.ROOT);
X.isRoot = jo;
var Qo = ke.bind(null, se.SELECTOR);
X.isSelector = Qo;
var Yo = ke.bind(null, se.STRING);
X.isString = Yo;
var Lr = ke.bind(null, se.TAG);
X.isTag = Lr;
var Vo = ke.bind(null, se.UNIVERSAL);
X.isUniversal = Vo;
function Fr(a) {
  return Yt(a) && a.value && (a.value.startsWith("::") || a.value.toLowerCase() === ":before" || a.value.toLowerCase() === ":after" || a.value.toLowerCase() === ":first-letter" || a.value.toLowerCase() === ":first-line");
}
function Ko(a) {
  return Yt(a) && !Fr(a);
}
function Jo(a) {
  return !!(Qt(a) && a.walk);
}
function Xo(a) {
  return Dr(a) || Lr(a);
}
(function(a) {
  a.__esModule = !0;
  var t = G;
  Object.keys(t).forEach(function(v) {
    v === "default" || v === "__esModule" || v in a && a[v] === t[v] || (a[v] = t[v]);
  });
  var b = re;
  Object.keys(b).forEach(function(v) {
    v === "default" || v === "__esModule" || v in a && a[v] === b[v] || (a[v] = b[v]);
  });
  var p = X;
  Object.keys(p).forEach(function(v) {
    v === "default" || v === "__esModule" || v in a && a[v] === p[v] || (a[v] = p[v]);
  });
})(Br);
(function(a, t) {
  t.__esModule = !0, t.default = void 0;
  var b = f(go), p = y(Br);
  function v() {
    if (typeof WeakMap != "function")
      return null;
    var o = /* @__PURE__ */ new WeakMap();
    return v = function() {
      return o;
    }, o;
  }
  function y(o) {
    if (o && o.__esModule)
      return o;
    if (o === null || typeof o != "object" && typeof o != "function")
      return { default: o };
    var l = v();
    if (l && l.has(o))
      return l.get(o);
    var c = {}, w = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var B in o)
      if (Object.prototype.hasOwnProperty.call(o, B)) {
        var F = w ? Object.getOwnPropertyDescriptor(o, B) : null;
        F && (F.get || F.set) ? Object.defineProperty(c, B, F) : c[B] = o[B];
      }
    return c.default = o, l && l.set(o, c), c;
  }
  function f(o) {
    return o && o.__esModule ? o : { default: o };
  }
  var g = function(l) {
    return new b.default(l);
  };
  Object.assign(g, p), delete g.__esModule;
  var n = g;
  t.default = n, a.exports = t.default;
})(wt, wt.exports);
var Zo = wt.exports;
const ea = ui, qr = Zo, ta = qr();
var ra = {
  isUsableColor(a, t) {
    return ea(t) && a !== "gray" && t[600];
  },
  /**
   * @param {string} selector
   */
  commonTrailingPseudos(a) {
    let t = ta.astSync(a), b = [];
    for (let [v, y] of t.nodes.entries())
      for (const [f, g] of [...y.nodes].reverse().entries()) {
        if (g.type !== "pseudo" || !g.value.startsWith("::"))
          break;
        b[f] = b[f] || [], b[f][v] = g;
      }
    let p = qr.selector();
    for (const v of b) {
      if (!v)
        continue;
      if ((/* @__PURE__ */ new Set([...v.map((f) => f.value)])).size > 1)
        break;
      v.forEach((f) => f.remove()), p.prepend(v[0]);
    }
    return p.nodes.length ? [p.toString(), t.toString()] : [null, a];
  }
};
const na = Hn, ia = Gn, oa = Yn, aa = Jn, { commonTrailingPseudos: sa } = ra, ur = {
  // Reserved for future "magic properties", for example:
  // bulletColor: (color) => ({ 'ul > li::before': { backgroundColor: color } }),
};
function zt(a, { className: t, modifier: b, prefix: p }) {
  let v = p(`.not-${t}`).slice(1), y = a.startsWith(">") ? `${b === "DEFAULT" ? `.${t}` : `.${t}-${b}`} ` : "", [f, g] = sa(a);
  return f ? `:where(${y}${g}):not(:where([class~="${v}"],[class~="${v}"] *))${f}` : `:where(${y}${a}):not(:where([class~="${v}"],[class~="${v}"] *))`;
}
function lr(a) {
  return typeof a == "object" && a !== null;
}
function ua(a = {}, { target: t, className: b, modifier: p, prefix: v }) {
  function y(f, g) {
    return t === "legacy" ? [f, g] : Array.isArray(g) ? [f, g] : lr(g) ? Object.values(g).some(lr) ? [
      zt(f, { className: b, modifier: p, prefix: v }),
      g,
      Object.fromEntries(Object.entries(g).map(([o, l]) => y(o, l)))
    ] : [zt(f, { className: b, modifier: p, prefix: v }), g] : [f, g];
  }
  return Object.fromEntries(
    Object.entries(
      ia(
        {},
        ...Object.keys(a).filter((f) => ur[f]).map((f) => ur[f](a[f])),
        ...oa(a.css || {})
      )
    ).map(([f, g]) => y(f, g))
  );
}
var la = na.withOptions(
  ({ className: a = "prose", target: t = "modern" } = {}) => function({ addVariant: b, addComponents: p, theme: v, prefix: y }) {
    let f = v("typography"), g = { className: a, prefix: y };
    for (let [n, ...o] of [
      ["headings", "h1", "h2", "h3", "h4", "h5", "h6", "th"],
      ["h1"],
      ["h2"],
      ["h3"],
      ["h4"],
      ["h5"],
      ["h6"],
      ["p"],
      ["a"],
      ["blockquote"],
      ["figure"],
      ["figcaption"],
      ["strong"],
      ["em"],
      ["kbd"],
      ["code"],
      ["pre"],
      ["ol"],
      ["ul"],
      ["li"],
      ["table"],
      ["thead"],
      ["tr"],
      ["th"],
      ["td"],
      ["img"],
      ["video"],
      ["hr"],
      ["lead", '[class~="lead"]']
    ]) {
      o = o.length === 0 ? [n] : o;
      let l = t === "legacy" ? o.map((c) => `& ${c}`) : o.join(", ");
      b(
        `${a}-${n}`,
        t === "legacy" ? l : `& :is(${zt(l, g)})`
      );
    }
    p(
      Object.keys(f).map((n) => ({
        [n === "DEFAULT" ? `.${a}` : `.${a}-${n}`]: ua(
          f[n],
          {
            target: t,
            className: a,
            modifier: n,
            prefix: y
          }
        )
      }))
    );
  },
  () => ({
    theme: { typography: aa }
  })
);
const ca = /* @__PURE__ */ $n(la), da = {
  content: [],
  future: {
    hoverOnlyWhenSupported: !0
  },
  theme: {
    extend: {
      typography: ({ theme: a }) => ({
        DEFAULT: {
          css: [
            {
              color: a("colors.neutral.900"),
              fontSize: "16px",
              fontFamily: a("fontFamily.body"),
              p: {
                marginTop: "20px",
                marginBottom: "20px"
              },
              '[class~="lead"]': {
                fontWeight: a("fontWeight.medium"),
                fontSize: "20px",
                lineHeight: "32px",
                marginTop: "24px",
                marginBottom: "24px",
                color: a("colors.neutral.900")
              },
              blockquote: {
                fontWeight: a("fontWeight.medium"),
                color: a("colors.neutral.900"),
                marginTop: "32px",
                marginBottom: "32px",
                paddingLeft: "20px"
              },
              "blockquote ~ figcaption ": {
                fontWeight: a("fontWeight.medium"),
                color: a("colors.neutral.900"),
                fontStyle: "italic"
              },
              h1: {
                fontSize: "36px",
                fontFamily: a("fontFamily.headings"),
                marginTop: "0",
                marginBottom: "32px",
                lineHeight: "40px"
              },
              h2: {
                fontSize: "24px",
                fontFamily: a("fontFamily.headings"),
                marginTop: "48px",
                marginBottom: "24px",
                lineHeight: "32px"
              },
              h3: {
                fontSize: "20px",
                fontFamily: a("fontFamily.headings"),
                marginTop: "32px",
                marginBottom: "12px",
                lineHeight: "32px"
              },
              h4: {
                fontFamily: a("fontFamily.headings"),
                marginTop: "24px",
                marginBottom: "8px",
                lineHeight: "24px"
              },
              img: {
                marginTop: "32px",
                marginBottom: "32px"
              },
              video: {
                marginTop: "32px",
                marginBottom: "32px"
              },
              figure: {
                marginTop: "32px",
                marginBottom: "32px"
              },
              "figure > *": {
                marginTop: "0",
                marginBottom: "0"
              },
              figcaption: {
                fontWeight: a("fontStyle.italic"),
                fontSize: "14px",
                lineHeight: "20px",
                marginTop: "12px"
              },
              code: {
                fontSize: "14px"
              },
              "h2 code": {
                fontSize: "21px"
              },
              "h3 code": {
                fontSize: "18px"
              },
              strong: {
                fontWeight: a("fontWeight.medium")
              },
              pre: {
                fontSize: "14px",
                lineHeight: "24px",
                marginTop: "24px",
                marginBottom: "24px",
                borderRadius: "6px",
                paddingTop: "12px",
                paddingRight: "16px",
                paddingBottom: "12px",
                paddingLeft: "16px"
              },
              ol: {
                marginTop: "20px",
                marginBottom: "20px",
                paddingLeft: "20px"
              },
              ul: {
                marginTop: "20px",
                marginBottom: "20px",
                paddingLeft: "26px"
              },
              li: {
                marginTop: "8px",
                marginBottom: "8px"
              },
              "ol > li": {
                paddingLeft: "6px"
              },
              "ul > li": {
                paddingLeft: "6px"
              },
              "> ul > li p": {
                marginTop: "12px",
                marginBottom: "12px"
              },
              "> ul > li > *:first-child": {
                marginTop: "20px"
              },
              "> ul > li > *:last-child": {
                marginBottom: "20px"
              },
              "> ol > li > *:first-child": {
                marginTop: "20px"
              },
              "> ol > li > *:last-child": {
                marginBottom: "20px"
              },
              "ul ul, ul ol, ol ul, ol ol": {
                marginTop: "12px",
                marginBottom: "12px"
              },
              hr: {
                marginTop: "48px",
                marginBottom: "48px"
              },
              "hr + *": {
                marginTop: "0"
              },
              "h2 + *": {
                marginTop: "0"
              },
              "h3 + *": {
                marginTop: "0"
              },
              "h4 + *": {
                marginTop: "0"
              },
              table: {
                fontSize: "14px",
                lineHeight: "24px"
              },
              "thead th": {
                fontWeight: a("fontWeight.medium"),
                paddingRight: "8px",
                paddingBottom: "8px",
                paddingLeft: "8px"
              },
              "thead th:first-child": {
                paddingLeft: "0"
              },
              "thead th:last-child": {
                paddingRight: "0"
              },
              "tbody td, tfoot td": {
                paddingTop: "8px",
                paddingRight: "8px",
                paddingBottom: "8px",
                paddingLeft: "8px"
              },
              "tbody td:first-child, tfoot td:first-child": {
                paddingLeft: "0"
              },
              "tbody td:last-child, tfoot td:last-child": {
                paddingRight: "0"
              }
            },
            {
              "> :first-child": {
                marginTop: "0"
              },
              "> :last-child": {
                marginBottom: "0"
              }
            }
          ]
        }
      }),
      outlineColor: ({ theme: a }) => ({
        DEFAULT: a("colors.secondary.600")
      }),
      outlineOffset: {
        DEFAULT: "2px"
      },
      outlineWidth: {
        DEFAULT: "2px"
      },
      boxShadow: {
        DEFAULT: "0px 1px 3px rgba(0, 0, 0, 0.1), 0px 1px 2px rgba(0, 0, 0, 0.06)",
        md: "0px 4px 6px -1px rgba(0, 0, 0, 0.1), 0px 2px 4px -1px rgba(0, 0, 0, 0.06)"
      },
      backgroundImage: {
        "checked-checkbox-current": "linear-gradient(-45deg,transparent 65%, currentcolor 65.99%),linear-gradient(45deg,transparent 75%, currentcolor 75.99%),linear-gradient(-45deg, currentcolor 40%,transparent 40.99%),linear-gradient(45deg, currentcolor 30%, white 30.99%, white 40%,transparent 40.99%),linear-gradient(-45deg, white 50%, currentcolor 50.99%)",
        "indeterminate-checkbox-current": "linear-gradient(90deg,transparent 80%, currentcolor 80%),linear-gradient(-90deg,transparent 80%, currentcolor 80%),linear-gradient(0deg, currentcolor 43%, white 43%, white 57%, currentcolor 57%)"
      },
      colors: {
        brand: "rgb(var(--colors-brand) / <alpha-value>)",
        neutral: {
          900: "rgb(var(--colors-neutral-900) / <alpha-value>)",
          800: "rgb(var(--colors-neutral-800) / <alpha-value>)",
          700: "rgb(var(--colors-neutral-700) / <alpha-value>)",
          600: "rgb(var(--colors-neutral-600) / <alpha-value>)",
          500: "rgb(var(--colors-neutral-500) / <alpha-value>)",
          400: "rgb(var(--colors-neutral-400) / <alpha-value>)",
          300: "rgb(var(--colors-neutral-300) / <alpha-value>)",
          200: "rgb(var(--colors-neutral-200) / <alpha-value>)",
          100: "rgb(var(--colors-neutral-100) / <alpha-value>)",
          50: "rgb(var(--colors-neutral-50) / <alpha-value>)"
        },
        primary: {
          900: "rgb(var(--colors-primary-900) / <alpha-value>)",
          800: "rgb(var(--colors-primary-800) / <alpha-value>)",
          700: "rgb(var(--colors-primary-700) / <alpha-value>)",
          600: "rgb(var(--colors-primary-600) / <alpha-value>)",
          500: "rgb(var(--colors-primary-500) / <alpha-value>)",
          400: "rgb(var(--colors-primary-400) / <alpha-value>)",
          300: "rgb(var(--colors-primary-300) / <alpha-value>)",
          200: "rgb(var(--colors-primary-200) / <alpha-value>)",
          100: "rgb(var(--colors-primary-100) / <alpha-value>)",
          50: "rgb(var(--colors-primary-50) / <alpha-value>)"
        },
        secondary: {
          900: "rgb(var(--colors-secondary-900) / <alpha-value>)",
          800: "rgb(var(--colors-secondary-800) / <alpha-value>)",
          700: "rgb(var(--colors-secondary-700) / <alpha-value>)",
          600: "rgb(var(--colors-secondary-600) / <alpha-value>)",
          500: "rgb(var(--colors-secondary-500) / <alpha-value>)",
          400: "rgb(var(--colors-secondary-400) / <alpha-value>)",
          300: "rgb(var(--colors-secondary-300) / <alpha-value>)",
          200: "rgb(var(--colors-secondary-200) / <alpha-value>)",
          100: "rgb(var(--colors-secondary-100) / <alpha-value>)",
          50: "rgb(var(--colors-secondary-50) / <alpha-value>)"
        },
        positive: {
          900: "rgb(var(--colors-positive-900) / <alpha-value>)",
          800: "rgb(var(--colors-positive-800) / <alpha-value>)",
          700: "rgb(var(--colors-positive-700) / <alpha-value>)",
          600: "rgb(var(--colors-positive-600) / <alpha-value>)",
          500: "rgb(var(--colors-positive-500) / <alpha-value>)",
          400: "rgb(var(--colors-positive-400) / <alpha-value>)",
          300: "rgb(var(--colors-positive-300) / <alpha-value>)",
          200: "rgb(var(--colors-positive-200) / <alpha-value>)",
          100: "rgb(var(--colors-positive-100) / <alpha-value>)",
          50: "rgb(var(--colors-positive-50) / <alpha-value>)"
        },
        negative: {
          900: "rgb(var(--colors-negative-900) / <alpha-value>)",
          800: "rgb(var(--colors-negative-800) / <alpha-value>)",
          700: "rgb(var(--colors-negative-700) / <alpha-value>)",
          600: "rgb(var(--colors-negative-600) / <alpha-value>)",
          500: "rgb(var(--colors-negative-500) / <alpha-value>)",
          400: "rgb(var(--colors-negative-400) / <alpha-value>)",
          300: "rgb(var(--colors-negative-300) / <alpha-value>)",
          200: "rgb(var(--colors-negative-200) / <alpha-value>)",
          100: "rgb(var(--colors-negative-100) / <alpha-value>)",
          50: "rgb(var(--colors-negative-50) / <alpha-value>)"
        },
        warning: {
          900: "rgb(var(--colors-warning-900) / <alpha-value>)",
          800: "rgb(var(--colors-warning-800) / <alpha-value>)",
          700: "rgb(var(--colors-warning-700) / <alpha-value>)",
          600: "rgb(var(--colors-warning-600) / <alpha-value>)",
          500: "rgb(var(--colors-warning-500) / <alpha-value>)",
          400: "rgb(var(--colors-warning-400) / <alpha-value>)",
          300: "rgb(var(--colors-warning-300) / <alpha-value>)",
          200: "rgb(var(--colors-warning-200) / <alpha-value>)",
          100: "rgb(var(--colors-warning-100) / <alpha-value>)",
          50: "rgb(var(--colors-warning-50) / <alpha-value>)"
        },
        disabled: {
          900: "rgb(var(--colors-disabled-900) / 0.5)",
          800: "rgb(var(--colors-disabled-800) / 0.5)",
          700: "rgb(var(--colors-disabled-700) / 0.5)",
          600: "rgb(var(--colors-disabled-600) / 0.5)",
          500: "rgb(var(--colors-disabled-500) / 0.5)",
          400: "rgb(var(--colors-disabled-400) / 0.5)",
          300: "rgb(var(--colors-disabled-300) / 0.5)",
          200: "rgb(var(--colors-disabled-200) / 0.5)",
          100: "rgb(var(--colors-disabled-100) / 0.5)",
          50: "rgb(var(--colors-disabled-50) / 0.5)"
        }
      },
      fontFamily: {
        headings: "Inter, sans-serif",
        body: "Inter, sans-serif"
      },
      screens: {
        xs: "376px",
        sm: "640px"
      },
      keyframes: {
        "stroke-loader-circular": {
          "0%": { "stroke-dasharray": "1, 200", "stroke-dashoffset": "0" },
          "50%": { "stroke-dasharray": "140, 200", "stroke-dashoffset": "-35" },
          "100%": { "stroke-dasharray": "89, 200", "stroke-dashoffset": "-150" }
        },
        line: {
          from: {
            left: "-100%",
            width: "100%"
          },
          to: {
            left: "100%",
            width: "10%"
          }
        }
      },
      animation: {
        "spin-slow": "spin 1.5s linear infinite",
        line: "line 1.5s ease-in infinite",
        "stroke-loader-circular": "stroke-loader-circular 2s ease-in-out infinite"
      },
      fontSize: {
        "2xs": ["10px", "11px"],
        "3xs": ["8px", "8px"]
      }
    },
    variables: {
      DEFAULT: {
        colors: {
          primary: {
            50: "235 255 242",
            100: "217 253 228",
            200: "171 241 192",
            300: "130 234 158",
            400: "60 224 120",
            500: "19 195 96",
            600: "7 161 79",
            700: "13 127 63",
            800: "16 86 46",
            900: "15 50 30"
          },
          secondary: {
            50: "247 245 253",
            100: "239 236 251",
            200: "225 218 246",
            300: "216 203 245",
            400: "198 177 246",
            500: "180 151 249",
            600: "151 111 238",
            700: "119 79 209",
            800: "82 52 153",
            900: "48 32 86"
          },
          positive: {
            50: "235 255 242",
            100: "217 253 228",
            200: "171 241 192",
            300: "130 234 158",
            400: "60 224 120",
            500: "19 195 96",
            600: "7 161 79",
            700: "13 127 63",
            800: "16 86 46",
            900: "15 50 30"
          },
          negative: {
            50: "255 245 247",
            100: "255 232 237",
            200: "254 211 219",
            300: "253 193 202",
            400: "255 163 175",
            500: "255 127 143",
            600: "240 59 91",
            700: "208 13 48",
            800: "141 8 33",
            900: "76 15 23"
          },
          warning: {
            50: "254 247 236",
            100: "255 238 211",
            200: "254 220 165",
            300: "254 202 132",
            400: "255 181 77",
            500: "237 153 14",
            600: "191 121 17",
            700: "157 93 3",
            800: "109 63 9",
            900: "62 35 10"
          },
          neutral: {
            50: "249 251 250",
            100: "239 244 241",
            200: "217 226 220",
            300: "187 198 190",
            400: "129 140 133",
            500: "100 111 104",
            600: "77 86 79",
            700: "56 65 59",
            800: "37 43 39",
            900: "21 26 22"
          },
          disabled: {
            50: "249 251 250",
            100: "239 244 241",
            200: "217 226 220",
            300: "187 198 190",
            400: "129 140 133",
            500: "100 111 104",
            600: "77 86 79",
            700: "56 65 59",
            800: "37 43 39",
            900: "21 26 22"
          }
        }
      }
    }
  },
  plugins: [ca, Wn, Un]
};
export {
  da as tailwindConfig
};
