// @storefront-ui/react/dist/components/SfRating/SfRating.mjs
import { jsxs as u, jsx as s } from "react/jsx-runtime";
import g from "classnames";
import { clamp as S, roundToNearest as b, SfRatingSize as e } from "@storefront-ui/shared";
import h from "../SfIcons/SfIconStarFilled.mjs";
import N from "../SfIcons/SfIconStarHalf.mjs";
import y from "../SfIcons/SfIconStar.mjs";
const w = {
  [e.xs]: "text-xs",
  [e.sm]: "text-sm",
  [e.base]: "text-base",
  [e.lg]: "text-lg",
  [e.xl]: "text-xl"
};
function $({
  size: n = e.base,
  max: t = 5,
  value: i = 0,
  halfIncrement: c,
  ariaLabel: d,
  className: f,
  ...p
}) {
  const l = S(b(i, c ? 0.5 : 1), 0, t), a = +!Number.isInteger(l), o = Math.ceil(l - a), x = t - o - a, m = d ?? `${i} out of ${t}`;
  return /* @__PURE__ */ u(
    "div",
    {
      role: "img",
      "aria-label": m,
      title: m,
      className: g("inline-flex items-center text-warning-500", w[n], f),
      "data-testid": "rating",
      ...p,
      children: [
        [...Array(o).keys()].map((r) => /* @__PURE__ */ s(h, { "aria-hidden": "true", className: "w-[1.5em] h-[1.5em]" }, r)),
        !!a && /* @__PURE__ */ s(N, { "aria-hidden": "true", className: "w-[1.5em] h-[1.5em]" }),
        [...Array(x).keys()].map((r) => /* @__PURE__ */ s(y, { "aria-hidden": "true", className: "text-disabled-500 w-[1.5em] h-[1.5em]" }, r))
      ]
    }
  );
}
export {
  $ as default
};
