// @storefront-ui/react/dist/components/SfChip/SfChip.mjs
import { jsxs as d, Fragment as m, jsx as g } from "react/jsx-runtime";
import u from "classnames";
import { forwardRef as b, useId as h } from "react";
import { SfChipSize as t } from "@storefront-ui/shared";
const x = (e, r, i, n) => {
  switch (e) {
    case t.sm:
      return r ? "px-1.5" : [i ? "pl-1.5" : "pl-3", n ? "pr-1.5" : "pr-3"];
    default:
      return r ? "px-2" : [i ? "pl-2" : "pl-4", n ? "pr-2" : "pr-4"];
  }
}, y = (e) => {
  switch (e) {
    case t.sm:
      return "text-sm py-1.5 gap-1.5";
    default:
      return "text-base py-2 gap-2";
  }
}, j = b(
  ({
    children: e,
    size: r = t.base,
    className: i,
    inputProps: n,
    square: c = !1,
    slotPrefix: a,
    slotSuffix: p,
    ...o
  }, l) => {
    const s = h();
    return /* @__PURE__ */ d(m, { children: [
      /* @__PURE__ */ g(
        "input",
        {
          id: s,
          ref: l,
          className: "absolute w-0 outline-none appearance-none peer",
          type: "checkbox",
          ...n
        }
      ),
      /* @__PURE__ */ d(
        "label",
        {
          htmlFor: s,
          className: u(
            "cursor-pointer ring-1 ring-neutral-200 ring-inset rounded-full inline-flex items-center transition duration-300 justify-center outline-offset-2 outline-secondary-600 peer-next-checked:ring-2 peer-next-checked:ring-primary-700 hover:bg-primary-100 peer-next-hover:ring-primary-200 active:bg-primary-200 peer-next-active:ring-primary-300 peer-next-disabled:cursor-not-allowed peer-next-disabled:bg-disabled-100 peer-next-disabled:opacity-50 peer-next-disabled:ring-1 peer-next-disabled:ring-disabled-200 peer-next-disabled:hover:ring-disabled-200 peer-next-checked:hover:ring-primary-700 peer-next-checked:active:ring-primary-700 peer-next-focus-visible:outline",
            y(r),
            x(r, c, a, p),
            i
          ),
          "data-testid": "chip",
          ...o,
          children: [
            a,
            e,
            p
          ]
        }
      )
    ] });
  }
);
export {
  j as default
};
