// @storefront-ui/react/dist/components/SfScrollable/SfScrollable.mjs
"use client";
import { jsxs as K, jsx as o } from "react/jsx-runtime";
import { useMemo as M, cloneElement as v } from "react";
import t from "classnames";
import { SfScrollableDirection as w, SfScrollableButtonsPlacement as r } from "@storefront-ui/shared";
import { useScrollable as Q } from "../../hooks/useScrollable/useScrollable.mjs";
import y from "../SfButton/SfButton.mjs";
import U from "../SfIcons/SfIconChevronLeft.mjs";
import V from "../SfIcons/SfIconChevronRight.mjs";
import { polymorphicForwardRef as W } from "../../shared/utils/index.mjs";
import { mergeRefs as X } from "../../shared/utils/refs.mjs";
const Y = "div", se = W(
  ({
    as: R,
    direction: i = w.horizontal,
    buttonsPlacement: a = r.block,
    activeIndex: m,
    reduceMotion: c,
    drag: u,
    onDragStart: b,
    onDragEnd: g,
    onScroll: p,
    onPrev: n,
    onNext: f,
    isActiveIndexCentered: h,
    className: C,
    wrapperClassName: j,
    prevDisabled: x,
    nextDisabled: z,
    buttonPrevAriaLabel: q = "Previous",
    buttonNextAriaLabel: F = "Next",
    style: Z,
    children: P,
    slotPreviousButton: S,
    slotNextButton: k,
    ...T
  }, H) => {
    const L = R || Y, e = i === w.horizontal, s = a === r.block, l = a === r.floating, O = M(
      () => ({
        direction: i,
        activeIndex: m,
        reduceMotion: c,
        drag: u,
        onDragStart: b,
        onDragEnd: g,
        onScroll: p,
        onPrev: n,
        onNext: f,
        isActiveIndexCentered: h
      }),
      [
        i,
        m,
        c,
        u,
        h,
        b,
        g,
        p,
        n,
        f
      ]
    ), { state: E, containerRef: G, getNextButtonProps: B, getPrevButtonProps: N } = Q(O);
    function I({ classNameButton: d }) {
      return S ? v(S, N({ disabled: x, onClick: n })) : /* @__PURE__ */ o(
        y,
        {
          ...N({
            square: !0,
            variant: "secondary",
            size: "lg",
            disabled: x,
            slotPrefix: /* @__PURE__ */ o(U, {}),
            "aria-label": q,
            className: t(
              d,
              l ? "disabled:hidden" : "disabled:!ring-disabled-300 disabled:!text-disabled-500"
            )
          })
        }
      );
    }
    function J({ classNameButton: d }) {
      return k ? v(k, B({ disabled: z, onClick: f })) : /* @__PURE__ */ o(
        y,
        {
          ...B({
            square: !0,
            variant: "secondary",
            size: "lg",
            disabled: z,
            slotPrefix: /* @__PURE__ */ o(V, {}),
            "aria-label": F,
            className: t(
              d,
              l ? "disabled:hidden" : "disabled:!ring-disabled-300 disabled:!text-disabled-500"
            )
          })
        }
      );
    }
    return /* @__PURE__ */ K(
      "div",
      {
        className: t(
          "items-center relative",
          e ? "flex" : "flex-col h-full inline-flex",
          j
        ),
        children: [
          a !== r.none && /* @__PURE__ */ o(
            I,
            {
              classNameButton: t("!rounded-full bg-white hidden md:block !ring-neutral-500 !text-neutral-500", {
                "mr-4": s && e,
                "mb-4 rotate-90": s && !e,
                "absolute left-4 z-10": l && e,
                "absolute top-4 rotate-90 z-10": l && !e
              })
            }
          ),
          /* @__PURE__ */ o(
            L,
            {
              className: t(C, "motion-safe:scroll-smooth", {
                "overflow-x-auto flex gap-4": e,
                "overflow-y-auto flex flex-col gap-4": !e,
                "cursor-grab": E.isDragged
              }),
              ref: X([G, H]),
              ...T,
              children: P
            }
          ),
          a !== r.none && /* @__PURE__ */ o(
            J,
            {
              classNameButton: t("!rounded-full bg-white hidden md:block !ring-neutral-500 !text-neutral-500", {
                "ml-4": s && e,
                "mt-4 rotate-90": s && !e,
                "absolute right-4 z-10": l && e,
                "absolute bottom-4 rotate-90 z-10": l && !e
              })
            }
          )
        ]
      }
    );
  }
);
export {
  se as default
};
