// @storefront-ui/react/dist/components/SfScrollable/SfScrollable.mjs
import { jsxs as J, jsx as o } from "react/jsx-runtime";
import { useMemo as K, cloneElement as v } from "react";
import l from "classnames";
import { SfScrollableDirection as w, SfScrollableButtonsPlacement as t } from "@storefront-ui/shared";
import { useScrollable as M } from "../../hooks/useScrollable/useScrollable.mjs";
import y from "../SfButton/SfButton.mjs";
import Q from "../SfIcons/SfIconChevronLeft.mjs";
import U from "../SfIcons/SfIconChevronRight.mjs";
import { polymorphicForwardRef as V } from "../../shared/utils/index.mjs";
import { mergeRefs as W } from "../../shared/utils/refs.mjs";
const X = "div", Y = V(
  ({
    as: R,
    direction: i = w.horizontal,
    buttonsPlacement: r = t.block,
    activeIndex: c,
    reduceMotion: d,
    drag: u,
    onDragStart: b,
    onDragEnd: g,
    onScroll: p,
    onPrev: n,
    onNext: f,
    isActiveIndexCentered: h,
    className: C,
    wrapperClassName: L,
    prevDisabled: x,
    nextDisabled: S,
    buttonPrevAriaLabel: j = "Previous",
    buttonNextAriaLabel: q = "Next",
    style: Z,
    children: F,
    slotPreviousButton: z,
    slotNextButton: k,
    ...P
  }, T) => {
    const H = R || X, e = i === w.horizontal, a = r === t.block, s = r === t.floating, O = K(
      () => ({
        direction: i,
        activeIndex: c,
        reduceMotion: d,
        drag: u,
        onDragStart: b,
        onDragEnd: g,
        onScroll: p,
        onPrev: n,
        onNext: f,
        isActiveIndexCentered: h
      }),
      [
        i,
        c,
        d,
        u,
        h,
        b,
        g,
        p,
        n,
        f
      ]
    ), { state: $, containerRef: E, getNextButtonProps: B, getPrevButtonProps: N } = M(O);
    function G({ classNameButton: m }) {
      return z ? v(z, N({ disabled: x, onClick: n })) : /* @__PURE__ */ o(
        y,
        {
          ...N({
            square: !0,
            variant: "secondary",
            size: "lg",
            disabled: x,
            slotPrefix: /* @__PURE__ */ o(Q, {}),
            ariaLabel: j,
            className: l(
              "hidden md:block !ring-neutral-500 !text-neutral-500 disabled:!ring-disabled-300 disabled:!text-disabled-500",
              m
            )
          })
        }
      );
    }
    function I({ classNameButton: m }) {
      return k ? v(k, B({ disabled: S, onClick: f })) : /* @__PURE__ */ o(
        y,
        {
          ...B({
            square: !0,
            variant: "secondary",
            size: "lg",
            disabled: S,
            slotPrefix: /* @__PURE__ */ o(U, {}),
            ariaLabel: q,
            className: l(
              "hidden md:block !ring-neutral-500 !text-neutral-500 disabled:!ring-disabled-300 disabled:!text-disabled-500",
              m
            )
          })
        }
      );
    }
    return /* @__PURE__ */ J(
      "div",
      {
        className: l(
          "items-center relative",
          e ? "flex" : "flex-col h-full inline-flex",
          L
        ),
        children: [
          r !== t.none && /* @__PURE__ */ o(
            G,
            {
              classNameButton: l("!rounded-full bg-white", {
                "mr-4": a && e,
                "mb-4 rotate-90": a && !e,
                "absolute left-4 z-10": s && e,
                "absolute top-4 rotate-90 z-10": s && !e
              })
            }
          ),
          /* @__PURE__ */ o(
            H,
            {
              className: l(C, "motion-safe:scroll-smooth", {
                "overflow-x-auto flex gap-4": e,
                "overflow-y-auto flex flex-col gap-4": !e,
                "cursor-grab": $.isDragged
              }),
              ref: W([E, T]),
              ...P,
              children: F
            }
          ),
          r !== t.none && /* @__PURE__ */ o(
            I,
            {
              classNameButton: l("!rounded-full bg-white", {
                "ml-4": a && e,
                "mt-4 rotate-90": a && !e,
                "absolute right-4 z-10": s && e,
                "absolute bottom-4 rotate-90 z-10": s && !e
              })
            }
          )
        ]
      }
    );
  }
), ie = Y;
export {
  ie as default
};
