// @storefront-ui/react/dist/components/SfProgressCircular/SfProgressCircular.mjs
import { jsxs as x, jsx as g } from "react/jsx-runtime";
import t from "classnames";
import { SfProgressSize as r } from "@storefront-ui/shared";
const m = {
  [r.xs]: "h-4 w-4 ring-2",
  [r.sm]: "h-5 w-5 ring-2",
  [r.base]: "h-6 w-6 ring-2",
  [r.lg]: "h-8 w-8 ring-2",
  [r.xl]: "h-10 w-10 ring-2",
  [r["2xl"]]: "h-14 w-14 ring-[3px]",
  [r["3xl"]]: "h-24 w-24 ring-4",
  [r["4xl"]]: "h-48 w-48 ring-8"
}, c = {
  [r.xs]: "stroke-[10px]",
  [r.sm]: "stroke-[8px]",
  [r.base]: "stroke-[6px]",
  [r.lg]: "stroke-[4px]",
  [r.xl]: "stroke-[3px]",
  [r["2xl"]]: "stroke-[3px]",
  [r["3xl"]]: "stroke-2",
  [r["4xl"]]: "stroke-2"
};
function u({
  value: s = 0,
  size: e = r.base,
  ariaLabel: a = "Progress element",
  children: o,
  className: i,
  ...n
}) {
  const l = `${s / 100 * 151}, 150`;
  return /* @__PURE__ */ x(
    "svg",
    {
      role: "progressbar",
      "aria-valuemin": 0,
      "aria-valuemax": 100,
      "aria-valuenow": s,
      "aria-label": a,
      className: t(
        "inline-block ring-inset ring-neutral-300 text-primary-700 rounded-full transition-[stroke-dasharray] ease-in-out duration-500 text-sm",
        m[e],
        i
      ),
      viewBox: "25 25 50 50",
      strokeDasharray: l,
      "data-testid": "progress",
      ...n,
      children: [
        /* @__PURE__ */ g(
          "circle",
          {
            className: t("origin-bottom-right -rotate-90 stroke-current fill-none", c[e]),
            cx: "50",
            cy: "50",
            r: "24"
          }
        ),
        o
      ]
    }
  );
}
export {
  u as default
};
