// @storefront-ui/react/dist/components/SfDrawer/SfDrawer.mjs
"use client";
import { jsx as n } from "react/jsx-runtime";
import x from "classnames";
import { useRef as y } from "react";
import { useClickAway as D } from "react-use";
import { mergeRefs as R } from "../../shared/utils/refs.mjs";
import { SfDrawerPlacement as f } from "@storefront-ui/shared";
import { polymorphicForwardRef as b } from "../../shared/utils/index.mjs";
const h = "aside", k = (r) => ({
  "left-0": r !== f.right,
  "right-0": r !== f.left,
  "top-0": r !== f.bottom,
  "bottom-0": r !== f.top
}), F = b(
  ({
    as: r,
    open: m,
    placement: a = f.left,
    disableClickAway: i,
    disableEsc: o,
    onClose: t,
    className: s,
    children: p,
    ...c
  }, d) => {
    const w = r || h, e = y(null);
    D(e, () => {
      i || t == null || t();
    });
    const u = (g) => {
      o || g.key === "Escape" && (t == null || t());
    };
    return m ? /* @__PURE__ */ n(
      w,
      {
        ref: R([e, d]),
        className: x("fixed", k(a), s),
        tabIndex: "-1",
        "data-testid": "drawer",
        ...c,
        onKeyDown: u,
        children: p
      }
    ) : null;
  }
);
export {
  F as default
};
