// @storefront-ui/react/dist/components/SfListItem/SfListItem.mjs
import { jsxs as b, jsx as l } from "react/jsx-runtime";
import g from "classnames";
import { SfListItemSize as s } from "@storefront-ui/shared";
import { polymorphicForwardRef as h } from "../../shared/utils/index.mjs";
const v = {
  [s.sm]: "text-sm px-4 py-1",
  [s.base]: "px-4 py-2",
  [s.lg]: "p-4"
}, N = "li", w = "span", C = h((o, r) => {
  const {
    size: n = s.base,
    disabled: e,
    selected: c,
    className: f,
    slotPrefix: i,
    slotSuffix: a,
    as: m,
    childrenTag: d,
    children: u,
    ...p
  } = o, x = m || N, t = d || w;
  return /* @__PURE__ */ b(
    x,
    {
      ref: r,
      className: g(
        "inline-flex items-center gap-2 w-full hover:bg-neutral-100 active:bg-neutral-200 cursor-pointer focus-visible:outline focus-visible:outline-offset focus-visible:relative focus-visible:z-10",
        {
          "cursor-not-allowed pointer-events-none text-disabled-900": e,
          "font-medium": c
        },
        v[n],
        f
      ),
      disabled: e,
      "data-testid": "list-item",
      ...p,
      children: [
        i && /* @__PURE__ */ l(t, { className: e ? "text-disabled-500" : "text-neutral-500", children: i }),
        /* @__PURE__ */ l(t, { className: "flex flex-col w-full min-w-0", children: u }),
        a && /* @__PURE__ */ l(t, { className: e ? "text-disabled-500" : "text-neutral-500", children: a })
      ]
    }
  );
});
export {
  C as default
};
