// @storefront-ui/react/dist/components/SfListItem/SfListItem.mjs
import { jsxs as u, jsx as t } from "react/jsx-runtime";
import p from "classnames";
import { SfListItemSize as s } from "@storefront-ui/shared";
import { polymorphicForwardRef as x } from "../../shared/utils/index.mjs";
const b = {
  [s.sm]: "text-sm px-4 py-1",
  [s.base]: "px-4 py-2",
  [s.lg]: "p-4"
}, v = "li", g = x((a, o) => {
  const {
    size: r = s.base,
    disabled: e,
    selected: n,
    className: c,
    slotPrefix: i,
    slotSuffix: l,
    as: m,
    children: f,
    ...d
  } = a;
  return /* @__PURE__ */ u(
    m || v,
    {
      ref: o,
      className: p(
        "inline-flex items-center gap-2 w-full hover:bg-neutral-100 active:bg-neutral-200 cursor-pointer focus-visible:outline focus-visible:outline-offset focus-visible:relative focus-visible:z-10",
        {
          "cursor-not-allowed pointer-events-none text-disabled-900": e,
          "font-medium": n
        },
        b[r],
        c
      ),
      disabled: e,
      "data-testid": "list-item",
      ...d,
      children: [
        i && /* @__PURE__ */ t("span", { className: e ? "text-disabled-500" : "text-neutral-500", children: i }),
        /* @__PURE__ */ t("span", { className: "flex flex-col w-full min-w-0", children: f }),
        l && /* @__PURE__ */ t("span", { className: e ? "text-disabled-500" : "text-neutral-500", children: l })
      ]
    }
  );
}), I = g;
export {
  I as default
};
