// @storefront-ui/react/dist/components/SfRatingButton/SfRatingButton.mjs
import { jsx as s, jsxs as N } from "react/jsx-runtime";
import { useState as y } from "react";
import M from "classnames";
import { SfRatingButtonSize as r, SfIconSize as n } from "@storefront-ui/shared";
import F from "../SfIcons/SfIconStarFilled.mjs";
import H from "../SfIcons/SfIconStar.mjs";
function g(t) {
  return `${t} Star${t !== 1 ? "s" : ""}`;
}
const j = ({ isFilled: t, iconSize: o }) => t ? /* @__PURE__ */ s(
  F,
  {
    role: "none",
    className: "text-primary-700 cursor-pointer peer-disabled:cursor-default peer-disabled:text-disabled-500 peer-focus-visible:outline",
    size: o
  }
) : /* @__PURE__ */ s(
  H,
  {
    "data-testid": "ratingbutton-icon-empty",
    role: "none",
    className: "text-neutral-500 cursor-pointer peer-disabled:cursor-default peer-disabled:text-disabled-500 peer-focus-visible:outline",
    size: o
  }
), B = {
  [r.sm]: n.base,
  [r.base]: n.lg,
  [r.lg]: n.xl
};
function E({
  onChange: t,
  value: o = 0,
  disabled: i = !1,
  max: a = 5,
  name: d = "sf-rating-button",
  className: f,
  size: u = r.base,
  getLabelText: m = g,
  children: p = j,
  ...b
}) {
  const [l, c] = y(0), S = Array.from({ length: Math.floor(Math.abs(a)) }, (e, I) => I + 1), x = (e) => e <= l || l === 0 && e <= o, h = (e) => {
    t == null || t(Number(e.target.value));
  }, v = (e) => () => {
    i || c(e);
  }, z = () => {
    i || c(0);
  };
  return /* @__PURE__ */ s("div", { role: "radiogroup", className: M("flex", f), "data-testid": "ratingbutton", ...b, children: S.map((e) => /* @__PURE__ */ N("label", { onMouseEnter: v(e), onMouseLeave: z, children: [
    /* @__PURE__ */ s(
      "input",
      {
        type: "radio",
        name: d,
        value: e,
        checked: e === o,
        onChange: h,
        disabled: i,
        "aria-label": m(e),
        className: "sr-only peer"
      }
    ),
    p({ isFilled: x(e), max: a, iconSize: B[u] })
  ] }, e)) });
}
export {
  E as default
};
