// @storefront-ui/react/dist/components/SfBadge/SfBadge.mjs
import { jsx as p } from "react/jsx-runtime";
import { SfBadgePlacement as m } from "@storefront-ui/shared";
import x from "classnames";
function b({
  content: a,
  variant: s,
  max: r = 99,
  placement: t = m["top-right"],
  className: l,
  ...i
}) {
  const e = s === "dot";
  let o = a;
  return e ? o = "" : !Number.isNaN(a) && Number(a) > r && (o = `${r}+`), /* @__PURE__ */ p(
    "span",
    {
      className: x(
        "block absolute py-0.5 px-1 bg-secondary-700 font-medium text-white text-[8px] leading-[8px] rounded-xl",
        {
          "min-w-[12px] min-h-[12px]": !e,
          "w-[10px] h-[10px]": e,
          "top-0 right-0 -translate-x-0.5 translate-y-0.5": t === "top-right",
          "top-0 left-0 translate-x-0.5 translate-y-0.5": t === "top-left",
          "bottom-0 right-0 -translate-x-0.5 -translate-y-0.5": t === "bottom-right",
          "bottom-0 left-0 translate-x-0.5 -translate-y-0.5": t === "bottom-left"
        },
        l
      ),
      "data-testid": "badge",
      ...i,
      children: o
    }
  );
}
export {
  b as default
};
